// @pulumi/cloudflare/accountToken.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountToken = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/accountToken:AccountToken example '<account_id>/<token_id>'
 * ```
 */
class AccountToken extends pulumi.CustomResource {
    /**
     * Get an existing AccountToken resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AccountToken(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of AccountToken.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccountToken.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["condition"] = state?.condition;
            resourceInputs["expiresOn"] = state?.expiresOn;
            resourceInputs["issuedOn"] = state?.issuedOn;
            resourceInputs["lastUsedOn"] = state?.lastUsedOn;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["name"] = state?.name;
            resourceInputs["notBefore"] = state?.notBefore;
            resourceInputs["policies"] = state?.policies;
            resourceInputs["status"] = state?.status;
            resourceInputs["value"] = state?.value;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.policies === undefined && !opts.urn) {
                throw new Error("Missing required property 'policies'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["condition"] = args?.condition;
            resourceInputs["expiresOn"] = args?.expiresOn;
            resourceInputs["name"] = args?.name;
            resourceInputs["notBefore"] = args?.notBefore;
            resourceInputs["policies"] = args?.policies;
            resourceInputs["status"] = args?.status;
            resourceInputs["issuedOn"] = undefined /*out*/;
            resourceInputs["lastUsedOn"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
            resourceInputs["value"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["value"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(AccountToken.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AccountToken = AccountToken;
/** @internal */
AccountToken.__pulumiType = 'cloudflare:index/accountToken:AccountToken';
