// @pulumi/cloudflare/accountSubscription.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.AccountSubscription = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleAccountSubscription = new cloudflare.AccountSubscription("example_account_subscription", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     frequency: "monthly",
 *     ratePlan: {
 *         id: "free",
 *         currency: "USD",
 *         externallyManaged: false,
 *         isContract: false,
 *         publicName: "Business Plan",
 *         scope: "zone",
 *         sets: ["string"],
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/accountSubscription:AccountSubscription example '<account_id>'
 * ```
 */
class AccountSubscription extends pulumi.CustomResource {
    /**
     * Get an existing AccountSubscription resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new AccountSubscription(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of AccountSubscription.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === AccountSubscription.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["currency"] = state?.currency;
            resourceInputs["currentPeriodEnd"] = state?.currentPeriodEnd;
            resourceInputs["currentPeriodStart"] = state?.currentPeriodStart;
            resourceInputs["frequency"] = state?.frequency;
            resourceInputs["price"] = state?.price;
            resourceInputs["ratePlan"] = state?.ratePlan;
            resourceInputs["state"] = state?.state;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["frequency"] = args?.frequency;
            resourceInputs["ratePlan"] = args?.ratePlan;
            resourceInputs["currency"] = undefined /*out*/;
            resourceInputs["currentPeriodEnd"] = undefined /*out*/;
            resourceInputs["currentPeriodStart"] = undefined /*out*/;
            resourceInputs["price"] = undefined /*out*/;
            resourceInputs["state"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(AccountSubscription.__pulumiType, name, resourceInputs, opts);
    }
}
exports.AccountSubscription = AccountSubscription;
/** @internal */
AccountSubscription.__pulumiType = 'cloudflare:index/accountSubscription:AccountSubscription';
