// @pulumi/cloudflare/botManagement.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.BotManagement = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleBotManagement = new cloudflare.BotManagement("example_bot_management", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     aiBotsProtection: "block",
 *     cfRobotsVariant: "policy_only",
 *     crawlerProtection: "enabled",
 *     enableJs: true,
 *     fightMode: true,
 *     isRobotsTxtManaged: false,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/botManagement:BotManagement example '<zone_id>'
 * ```
 */
class BotManagement extends pulumi.CustomResource {
    /**
     * Get an existing BotManagement resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new BotManagement(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of BotManagement.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === BotManagement.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["aiBotsProtection"] = state?.aiBotsProtection;
            resourceInputs["autoUpdateModel"] = state?.autoUpdateModel;
            resourceInputs["bmCookieEnabled"] = state?.bmCookieEnabled;
            resourceInputs["cfRobotsVariant"] = state?.cfRobotsVariant;
            resourceInputs["crawlerProtection"] = state?.crawlerProtection;
            resourceInputs["enableJs"] = state?.enableJs;
            resourceInputs["fightMode"] = state?.fightMode;
            resourceInputs["isRobotsTxtManaged"] = state?.isRobotsTxtManaged;
            resourceInputs["optimizeWordpress"] = state?.optimizeWordpress;
            resourceInputs["sbfmDefinitelyAutomated"] = state?.sbfmDefinitelyAutomated;
            resourceInputs["sbfmLikelyAutomated"] = state?.sbfmLikelyAutomated;
            resourceInputs["sbfmStaticResourceProtection"] = state?.sbfmStaticResourceProtection;
            resourceInputs["sbfmVerifiedBots"] = state?.sbfmVerifiedBots;
            resourceInputs["staleZoneConfiguration"] = state?.staleZoneConfiguration;
            resourceInputs["suppressSessionScore"] = state?.suppressSessionScore;
            resourceInputs["usingLatestModel"] = state?.usingLatestModel;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["aiBotsProtection"] = args?.aiBotsProtection;
            resourceInputs["autoUpdateModel"] = args?.autoUpdateModel;
            resourceInputs["bmCookieEnabled"] = args?.bmCookieEnabled;
            resourceInputs["cfRobotsVariant"] = args?.cfRobotsVariant;
            resourceInputs["crawlerProtection"] = args?.crawlerProtection;
            resourceInputs["enableJs"] = args?.enableJs;
            resourceInputs["fightMode"] = args?.fightMode;
            resourceInputs["isRobotsTxtManaged"] = args?.isRobotsTxtManaged;
            resourceInputs["optimizeWordpress"] = args?.optimizeWordpress;
            resourceInputs["sbfmDefinitelyAutomated"] = args?.sbfmDefinitelyAutomated;
            resourceInputs["sbfmLikelyAutomated"] = args?.sbfmLikelyAutomated;
            resourceInputs["sbfmStaticResourceProtection"] = args?.sbfmStaticResourceProtection;
            resourceInputs["sbfmVerifiedBots"] = args?.sbfmVerifiedBots;
            resourceInputs["suppressSessionScore"] = args?.suppressSessionScore;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["staleZoneConfiguration"] = undefined /*out*/;
            resourceInputs["usingLatestModel"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(BotManagement.__pulumiType, name, resourceInputs, opts);
    }
}
exports.BotManagement = BotManagement;
/** @internal */
BotManagement.__pulumiType = 'cloudflare:index/botManagement:BotManagement';
