// @pulumi/cloudflare/callsSfuApp.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CallsSfuApp = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleCallsSfuApp = new cloudflare.CallsSfuApp("example_calls_sfu_app", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     name: "production-realtime-app",
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class CallsSfuApp extends pulumi.CustomResource {
    /**
     * Get an existing CallsSfuApp resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CallsSfuApp(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of CallsSfuApp.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CallsSfuApp.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["appId"] = state?.appId;
            resourceInputs["created"] = state?.created;
            resourceInputs["modified"] = state?.modified;
            resourceInputs["name"] = state?.name;
            resourceInputs["secret"] = state?.secret;
            resourceInputs["uid"] = state?.uid;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["appId"] = args?.appId;
            resourceInputs["name"] = args?.name;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["modified"] = undefined /*out*/;
            resourceInputs["secret"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["secret"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(CallsSfuApp.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CallsSfuApp = CallsSfuApp;
/** @internal */
CallsSfuApp.__pulumiType = 'cloudflare:index/callsSfuApp:CallsSfuApp';
