// @pulumi/cloudflare/certificatePack.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CertificatePack = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * > Certificate packs are not able to be updated in place. If
 * you require a zero downtime rotation, you can create multiple
 * resources using a 2-phase change where you have both resources
 * live at once and you remove the old one once you've confirmed
 * the certificate is available.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleCertificatePack = new cloudflare.CertificatePack("example_certificate_pack", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     certificateAuthority: "lets_encrypt",
 *     hosts: [
 *         "example.com",
 *         "*.example.com",
 *         "www.example.com",
 *     ],
 *     type: "advanced",
 *     validationMethod: "txt",
 *     validityDays: 14,
 *     cloudflareBranding: false,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/certificatePack:CertificatePack example '<zone_id>/<certificate_pack_id>'
 * ```
 *
 * While supported, importing isn't recommended and it is advised to replace the
 *
 * certificate entirely instead.
 */
class CertificatePack extends pulumi.CustomResource {
    /**
     * Get an existing CertificatePack resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CertificatePack(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of CertificatePack.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CertificatePack.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["certificateAuthority"] = state?.certificateAuthority;
            resourceInputs["certificates"] = state?.certificates;
            resourceInputs["cloudflareBranding"] = state?.cloudflareBranding;
            resourceInputs["hosts"] = state?.hosts;
            resourceInputs["primaryCertificate"] = state?.primaryCertificate;
            resourceInputs["status"] = state?.status;
            resourceInputs["type"] = state?.type;
            resourceInputs["validationErrors"] = state?.validationErrors;
            resourceInputs["validationMethod"] = state?.validationMethod;
            resourceInputs["validationRecords"] = state?.validationRecords;
            resourceInputs["validityDays"] = state?.validityDays;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.certificateAuthority === undefined && !opts.urn) {
                throw new Error("Missing required property 'certificateAuthority'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            if (args?.validationMethod === undefined && !opts.urn) {
                throw new Error("Missing required property 'validationMethod'");
            }
            if (args?.validityDays === undefined && !opts.urn) {
                throw new Error("Missing required property 'validityDays'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["certificateAuthority"] = args?.certificateAuthority;
            resourceInputs["cloudflareBranding"] = args?.cloudflareBranding;
            resourceInputs["hosts"] = args?.hosts;
            resourceInputs["type"] = args?.type;
            resourceInputs["validationMethod"] = args?.validationMethod;
            resourceInputs["validityDays"] = args?.validityDays;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["certificates"] = undefined /*out*/;
            resourceInputs["primaryCertificate"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["validationErrors"] = undefined /*out*/;
            resourceInputs["validationRecords"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CertificatePack.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CertificatePack = CertificatePack;
/** @internal */
CertificatePack.__pulumiType = 'cloudflare:index/certificatePack:CertificatePack';
