// @pulumi/cloudflare/cloudforceOneRequestAsset.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CloudforceOneRequestAsset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleCloudforceOneRequestAsset = new cloudflare.CloudforceOneRequestAsset("example_cloudforce_one_request_asset", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     requestId: "f174e90a-fafe-4643-bbbc-4a0ed4fc8415",
 *     page: 0,
 *     perPage: 10,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/cloudforceOneRequestAsset:CloudforceOneRequestAsset example '<account_id>/<request_id>/<asset_id>'
 * ```
 */
class CloudforceOneRequestAsset extends pulumi.CustomResource {
    /**
     * Get an existing CloudforceOneRequestAsset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CloudforceOneRequestAsset(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of CloudforceOneRequestAsset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CloudforceOneRequestAsset.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["created"] = state?.created;
            resourceInputs["description"] = state?.description;
            resourceInputs["fileType"] = state?.fileType;
            resourceInputs["name"] = state?.name;
            resourceInputs["page"] = state?.page;
            resourceInputs["perPage"] = state?.perPage;
            resourceInputs["requestId"] = state?.requestId;
            resourceInputs["source"] = state?.source;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.page === undefined && !opts.urn) {
                throw new Error("Missing required property 'page'");
            }
            if (args?.perPage === undefined && !opts.urn) {
                throw new Error("Missing required property 'perPage'");
            }
            if (args?.requestId === undefined && !opts.urn) {
                throw new Error("Missing required property 'requestId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["page"] = args?.page;
            resourceInputs["perPage"] = args?.perPage;
            resourceInputs["requestId"] = args?.requestId;
            resourceInputs["source"] = args?.source;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["fileType"] = undefined /*out*/;
            resourceInputs["name"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CloudforceOneRequestAsset.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CloudforceOneRequestAsset = CloudforceOneRequestAsset;
/** @internal */
CloudforceOneRequestAsset.__pulumiType = 'cloudflare:index/cloudforceOneRequestAsset:CloudforceOneRequestAsset';
