// @pulumi/cloudflare/connectivityDirectoryService.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ConnectivityDirectoryService = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleConnectivityDirectoryService = new cloudflare.ConnectivityDirectoryService("example_connectivity_directory_service", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     host: {
 *         hostname: "api.example.com",
 *         resolverNetwork: {
 *             tunnelId: "0191dce4-9ab4-7fce-b660-8e5dec5172da",
 *             resolverIps: ["string"],
 *         },
 *     },
 *     name: "web-server",
 *     type: "http",
 *     httpPort: 8080,
 *     httpsPort: 8443,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/connectivityDirectoryService:ConnectivityDirectoryService example '<account_id>/<service_id>'
 * ```
 */
class ConnectivityDirectoryService extends pulumi.CustomResource {
    /**
     * Get an existing ConnectivityDirectoryService resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ConnectivityDirectoryService(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ConnectivityDirectoryService.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ConnectivityDirectoryService.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["host"] = state?.host;
            resourceInputs["httpPort"] = state?.httpPort;
            resourceInputs["httpsPort"] = state?.httpsPort;
            resourceInputs["name"] = state?.name;
            resourceInputs["serviceId"] = state?.serviceId;
            resourceInputs["type"] = state?.type;
            resourceInputs["updatedAt"] = state?.updatedAt;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.host === undefined && !opts.urn) {
                throw new Error("Missing required property 'host'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["host"] = args?.host;
            resourceInputs["httpPort"] = args?.httpPort;
            resourceInputs["httpsPort"] = args?.httpsPort;
            resourceInputs["name"] = args?.name;
            resourceInputs["type"] = args?.type;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["serviceId"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ConnectivityDirectoryService.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ConnectivityDirectoryService = ConnectivityDirectoryService;
/** @internal */
ConnectivityDirectoryService.__pulumiType = 'cloudflare:index/connectivityDirectoryService:ConnectivityDirectoryService';
