// @pulumi/cloudflare/customHostnameFallbackOrigin.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.CustomHostnameFallbackOrigin = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleCustomHostnameFallbackOrigin = new cloudflare.CustomHostnameFallbackOrigin("example_custom_hostname_fallback_origin", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     origin: "fallback.example.com",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/customHostnameFallbackOrigin:CustomHostnameFallbackOrigin example '<zone_id>'
 * ```
 */
class CustomHostnameFallbackOrigin extends pulumi.CustomResource {
    /**
     * Get an existing CustomHostnameFallbackOrigin resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new CustomHostnameFallbackOrigin(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of CustomHostnameFallbackOrigin.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === CustomHostnameFallbackOrigin.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["errors"] = state?.errors;
            resourceInputs["origin"] = state?.origin;
            resourceInputs["status"] = state?.status;
            resourceInputs["updatedAt"] = state?.updatedAt;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.origin === undefined && !opts.urn) {
                throw new Error("Missing required property 'origin'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["origin"] = args?.origin;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(CustomHostnameFallbackOrigin.__pulumiType, name, resourceInputs, opts);
    }
}
exports.CustomHostnameFallbackOrigin = CustomHostnameFallbackOrigin;
/** @internal */
CustomHostnameFallbackOrigin.__pulumiType = 'cloudflare:index/customHostnameFallbackOrigin:CustomHostnameFallbackOrigin';
