// @pulumi/cloudflare/dlpPredefinedProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DlpPredefinedProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustDlpPredefinedProfile = new cloudflare.ZeroTrustDlpPredefinedProfile("example_zero_trust_dlp_predefined_profile", {
 *     accountId: "account_id",
 *     profileId: "182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e",
 *     aiContextEnabled: true,
 *     allowedMatchCount: 5,
 *     confidenceThreshold: "confidence_threshold",
 *     enabledEntries: ["182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e"],
 *     entries: [{
 *         id: "182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e",
 *         enabled: true,
 *     }],
 *     ocrEnabled: true,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/dlpPredefinedProfile:DlpPredefinedProfile example '<account_id>/<profile_id>'
 * ```
 *
 * @deprecated cloudflare.index/dlppredefinedprofile.DlpPredefinedProfile has been deprecated in favor of cloudflare.index/zerotrustdlppredefinedprofile.ZeroTrustDlpPredefinedProfile
 */
class DlpPredefinedProfile extends pulumi.CustomResource {
    /**
     * Get an existing DlpPredefinedProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        pulumi.log.warn("DlpPredefinedProfile is deprecated: cloudflare.index/dlppredefinedprofile.DlpPredefinedProfile has been deprecated in favor of cloudflare.index/zerotrustdlppredefinedprofile.ZeroTrustDlpPredefinedProfile");
        return new DlpPredefinedProfile(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DlpPredefinedProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DlpPredefinedProfile.__pulumiType;
    }
    /** @deprecated cloudflare.index/dlppredefinedprofile.DlpPredefinedProfile has been deprecated in favor of cloudflare.index/zerotrustdlppredefinedprofile.ZeroTrustDlpPredefinedProfile */
    constructor(name, argsOrState, opts) {
        pulumi.log.warn("DlpPredefinedProfile is deprecated: cloudflare.index/dlppredefinedprofile.DlpPredefinedProfile has been deprecated in favor of cloudflare.index/zerotrustdlppredefinedprofile.ZeroTrustDlpPredefinedProfile");
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["aiContextEnabled"] = state?.aiContextEnabled;
            resourceInputs["allowedMatchCount"] = state?.allowedMatchCount;
            resourceInputs["confidenceThreshold"] = state?.confidenceThreshold;
            resourceInputs["enabledEntries"] = state?.enabledEntries;
            resourceInputs["entries"] = state?.entries;
            resourceInputs["name"] = state?.name;
            resourceInputs["ocrEnabled"] = state?.ocrEnabled;
            resourceInputs["openAccess"] = state?.openAccess;
            resourceInputs["profileId"] = state?.profileId;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.profileId === undefined && !opts.urn) {
                throw new Error("Missing required property 'profileId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["aiContextEnabled"] = args?.aiContextEnabled;
            resourceInputs["allowedMatchCount"] = args?.allowedMatchCount;
            resourceInputs["confidenceThreshold"] = args?.confidenceThreshold;
            resourceInputs["enabledEntries"] = args?.enabledEntries;
            resourceInputs["entries"] = args?.entries;
            resourceInputs["ocrEnabled"] = args?.ocrEnabled;
            resourceInputs["profileId"] = args?.profileId;
            resourceInputs["name"] = undefined /*out*/;
            resourceInputs["openAccess"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/dlpPredefinedProfile:DlpPredefinedProfile" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(DlpPredefinedProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DlpPredefinedProfile = DlpPredefinedProfile;
/** @internal */
DlpPredefinedProfile.__pulumiType = 'cloudflare:index/dlpPredefinedProfile:DlpPredefinedProfile';
