// @pulumi/cloudflare/dnsFirewall.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.DnsFirewall = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleDnsFirewall = new cloudflare.DnsFirewall("example_dns_firewall", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     name: "My Awesome DNS Firewall cluster",
 *     upstreamIps: [
 *         "192.0.2.1",
 *         "198.51.100.1",
 *         "2001:DB8:100::CF",
 *     ],
 *     attackMitigation: {
 *         enabled: true,
 *         onlyWhenUpstreamUnhealthy: false,
 *     },
 *     deprecateAnyRequests: true,
 *     ecsFallback: false,
 *     maximumCacheTtl: 900,
 *     minimumCacheTtl: 60,
 *     negativeCacheTtl: 900,
 *     ratelimit: 600,
 *     retries: 2,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/dnsFirewall:DnsFirewall example '<account_id>/<dns_firewall_id>'
 * ```
 */
class DnsFirewall extends pulumi.CustomResource {
    /**
     * Get an existing DnsFirewall resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new DnsFirewall(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of DnsFirewall.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === DnsFirewall.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["attackMitigation"] = state?.attackMitigation;
            resourceInputs["deprecateAnyRequests"] = state?.deprecateAnyRequests;
            resourceInputs["dnsFirewallIps"] = state?.dnsFirewallIps;
            resourceInputs["ecsFallback"] = state?.ecsFallback;
            resourceInputs["maximumCacheTtl"] = state?.maximumCacheTtl;
            resourceInputs["minimumCacheTtl"] = state?.minimumCacheTtl;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["name"] = state?.name;
            resourceInputs["negativeCacheTtl"] = state?.negativeCacheTtl;
            resourceInputs["ratelimit"] = state?.ratelimit;
            resourceInputs["retries"] = state?.retries;
            resourceInputs["upstreamIps"] = state?.upstreamIps;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.upstreamIps === undefined && !opts.urn) {
                throw new Error("Missing required property 'upstreamIps'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["attackMitigation"] = args?.attackMitigation;
            resourceInputs["deprecateAnyRequests"] = args?.deprecateAnyRequests;
            resourceInputs["ecsFallback"] = args?.ecsFallback;
            resourceInputs["maximumCacheTtl"] = args?.maximumCacheTtl;
            resourceInputs["minimumCacheTtl"] = args?.minimumCacheTtl;
            resourceInputs["name"] = args?.name;
            resourceInputs["negativeCacheTtl"] = args?.negativeCacheTtl;
            resourceInputs["ratelimit"] = args?.ratelimit;
            resourceInputs["retries"] = args?.retries;
            resourceInputs["upstreamIps"] = args?.upstreamIps;
            resourceInputs["dnsFirewallIps"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(DnsFirewall.__pulumiType, name, resourceInputs, opts);
    }
}
exports.DnsFirewall = DnsFirewall;
/** @internal */
DnsFirewall.__pulumiType = 'cloudflare:index/dnsFirewall:DnsFirewall';
