// @pulumi/cloudflare/emailRoutingDns.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailRoutingDns = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleEmailRoutingDns = new cloudflare.EmailRoutingDns("example_email_routing_dns", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     name: "example.net",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/emailRoutingDns:EmailRoutingDns example '<zone_id>'
 * ```
 */
class EmailRoutingDns extends pulumi.CustomResource {
    /**
     * Get an existing EmailRoutingDns resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EmailRoutingDns(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EmailRoutingDns.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EmailRoutingDns.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["created"] = state?.created;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["errors"] = state?.errors;
            resourceInputs["messages"] = state?.messages;
            resourceInputs["modified"] = state?.modified;
            resourceInputs["name"] = state?.name;
            resourceInputs["result"] = state?.result;
            resourceInputs["resultInfo"] = state?.resultInfo;
            resourceInputs["skipWizard"] = state?.skipWizard;
            resourceInputs["status"] = state?.status;
            resourceInputs["success"] = state?.success;
            resourceInputs["tag"] = state?.tag;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["name"] = args?.name;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["enabled"] = undefined /*out*/;
            resourceInputs["errors"] = undefined /*out*/;
            resourceInputs["messages"] = undefined /*out*/;
            resourceInputs["modified"] = undefined /*out*/;
            resourceInputs["result"] = undefined /*out*/;
            resourceInputs["resultInfo"] = undefined /*out*/;
            resourceInputs["skipWizard"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["success"] = undefined /*out*/;
            resourceInputs["tag"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EmailRoutingDns.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EmailRoutingDns = EmailRoutingDns;
/** @internal */
EmailRoutingDns.__pulumiType = 'cloudflare:index/emailRoutingDns:EmailRoutingDns';
