// @pulumi/cloudflare/emailRoutingCatchAll.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailRoutingCatchAll = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleEmailRoutingCatchAll = new cloudflare.EmailRoutingCatchAll("example_email_routing_catch_all", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     actions: [{
 *         type: "forward",
 *         value: ["destinationaddress@example.net"],
 *     }],
 *     matchers: [{
 *         type: "all",
 *     }],
 *     enabled: true,
 *     name: "Send to user@example.net rule.",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/emailRoutingCatchAll:EmailRoutingCatchAll example '<zone_id>'
 * ```
 */
class EmailRoutingCatchAll extends pulumi.CustomResource {
    /**
     * Get an existing EmailRoutingCatchAll resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EmailRoutingCatchAll(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EmailRoutingCatchAll.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EmailRoutingCatchAll.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["actions"] = state?.actions;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["matchers"] = state?.matchers;
            resourceInputs["name"] = state?.name;
            resourceInputs["tag"] = state?.tag;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.actions === undefined && !opts.urn) {
                throw new Error("Missing required property 'actions'");
            }
            if (args?.matchers === undefined && !opts.urn) {
                throw new Error("Missing required property 'matchers'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["actions"] = args?.actions;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["matchers"] = args?.matchers;
            resourceInputs["name"] = args?.name;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["tag"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EmailRoutingCatchAll.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EmailRoutingCatchAll = EmailRoutingCatchAll;
/** @internal */
EmailRoutingCatchAll.__pulumiType = 'cloudflare:index/emailRoutingCatchAll:EmailRoutingCatchAll';
