// @pulumi/cloudflare/emailRoutingRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.EmailRoutingRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleEmailRoutingRule = new cloudflare.EmailRoutingRule("example_email_routing_rule", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     actions: [{
 *         type: "forward",
 *         value: ["destinationaddress@example.net"],
 *     }],
 *     matchers: [{
 *         type: "literal",
 *         field: "to",
 *         value: "test@example.com",
 *     }],
 *     enabled: true,
 *     name: "Send to user@example.net rule.",
 *     priority: 0,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/emailRoutingRule:EmailRoutingRule example '<zone_id>/<rule_identifier>'
 * ```
 */
class EmailRoutingRule extends pulumi.CustomResource {
    /**
     * Get an existing EmailRoutingRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new EmailRoutingRule(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of EmailRoutingRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === EmailRoutingRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["actions"] = state?.actions;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["matchers"] = state?.matchers;
            resourceInputs["name"] = state?.name;
            resourceInputs["priority"] = state?.priority;
            resourceInputs["tag"] = state?.tag;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.actions === undefined && !opts.urn) {
                throw new Error("Missing required property 'actions'");
            }
            if (args?.matchers === undefined && !opts.urn) {
                throw new Error("Missing required property 'matchers'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["actions"] = args?.actions;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["matchers"] = args?.matchers;
            resourceInputs["name"] = args?.name;
            resourceInputs["priority"] = args?.priority;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["tag"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(EmailRoutingRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.EmailRoutingRule = EmailRoutingRule;
/** @internal */
EmailRoutingRule.__pulumiType = 'cloudflare:index/emailRoutingRule:EmailRoutingRule';
