// @pulumi/cloudflare/healthcheck.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Healthcheck = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleHealthcheck = new cloudflare.Healthcheck("example_healthcheck", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     address: "www.example.com",
 *     name: "server-1",
 *     checkRegions: [
 *         "WEU",
 *         "ENAM",
 *     ],
 *     consecutiveFails: 0,
 *     consecutiveSuccesses: 0,
 *     description: "Health check for www.example.com",
 *     httpConfig: {
 *         allowInsecure: true,
 *         expectedBody: "success",
 *         expectedCodes: [
 *             "2xx",
 *             "302",
 *         ],
 *         followRedirects: true,
 *         header: {
 *             Host: ["example.com"],
 *             "X-App-ID": ["abc123"],
 *         },
 *         method: "GET",
 *         path: "/health",
 *         port: 0,
 *     },
 *     interval: 0,
 *     retries: 0,
 *     suspended: true,
 *     tcpConfig: {
 *         method: "connection_established",
 *         port: 0,
 *     },
 *     timeout: 0,
 *     type: "HTTPS",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/healthcheck:Healthcheck example '<zone_id>/<healthcheck_id>'
 * ```
 */
class Healthcheck extends pulumi.CustomResource {
    /**
     * Get an existing Healthcheck resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Healthcheck(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Healthcheck.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Healthcheck.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["address"] = state?.address;
            resourceInputs["checkRegions"] = state?.checkRegions;
            resourceInputs["consecutiveFails"] = state?.consecutiveFails;
            resourceInputs["consecutiveSuccesses"] = state?.consecutiveSuccesses;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["description"] = state?.description;
            resourceInputs["failureReason"] = state?.failureReason;
            resourceInputs["httpConfig"] = state?.httpConfig;
            resourceInputs["interval"] = state?.interval;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["name"] = state?.name;
            resourceInputs["retries"] = state?.retries;
            resourceInputs["status"] = state?.status;
            resourceInputs["suspended"] = state?.suspended;
            resourceInputs["tcpConfig"] = state?.tcpConfig;
            resourceInputs["timeout"] = state?.timeout;
            resourceInputs["type"] = state?.type;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.address === undefined && !opts.urn) {
                throw new Error("Missing required property 'address'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["address"] = args?.address;
            resourceInputs["checkRegions"] = args?.checkRegions;
            resourceInputs["consecutiveFails"] = args?.consecutiveFails;
            resourceInputs["consecutiveSuccesses"] = args?.consecutiveSuccesses;
            resourceInputs["description"] = args?.description;
            resourceInputs["httpConfig"] = args?.httpConfig;
            resourceInputs["interval"] = args?.interval;
            resourceInputs["name"] = args?.name;
            resourceInputs["retries"] = args?.retries;
            resourceInputs["suspended"] = args?.suspended;
            resourceInputs["tcpConfig"] = args?.tcpConfig;
            resourceInputs["timeout"] = args?.timeout;
            resourceInputs["type"] = args?.type;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["failureReason"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Healthcheck.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Healthcheck = Healthcheck;
/** @internal */
Healthcheck.__pulumiType = 'cloudflare:index/healthcheck:Healthcheck';
