// @pulumi/cloudflare/hostnameTlsSetting.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.HostnameTlsSetting = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleHostnameTlsSetting = new cloudflare.HostnameTlsSetting("example_hostname_tls_setting", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     settingId: "ciphers",
 *     hostname: "app.example.com",
 *     value: [
 *         "ECDHE-RSA-AES128-GCM-SHA256",
 *         "AES128-GCM-SHA256",
 *     ],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/hostnameTlsSetting:HostnameTlsSetting example '<zone_id>/<setting_id>'
 * ```
 */
class HostnameTlsSetting extends pulumi.CustomResource {
    /**
     * Get an existing HostnameTlsSetting resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new HostnameTlsSetting(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of HostnameTlsSetting.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === HostnameTlsSetting.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["hostname"] = state?.hostname;
            resourceInputs["settingId"] = state?.settingId;
            resourceInputs["status"] = state?.status;
            resourceInputs["updatedAt"] = state?.updatedAt;
            resourceInputs["value"] = state?.value;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.hostname === undefined && !opts.urn) {
                throw new Error("Missing required property 'hostname'");
            }
            if (args?.settingId === undefined && !opts.urn) {
                throw new Error("Missing required property 'settingId'");
            }
            if (args?.value === undefined && !opts.urn) {
                throw new Error("Missing required property 'value'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["hostname"] = args?.hostname;
            resourceInputs["settingId"] = args?.settingId;
            resourceInputs["value"] = args?.value;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(HostnameTlsSetting.__pulumiType, name, resourceInputs, opts);
    }
}
exports.HostnameTlsSetting = HostnameTlsSetting;
/** @internal */
HostnameTlsSetting.__pulumiType = 'cloudflare:index/hostnameTlsSetting:HostnameTlsSetting';
