// @pulumi/cloudflare/listItem.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ListItem = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleListItem = new cloudflare.ListItem("example_list_item", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     listId: "2c0fc9fa937b11eaa1b71c4d701ab86e",
 *     ip: "10.0.0.1",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/listItem:ListItem example '<account_id>/<list_id>/<item_id>'
 * ```
 *
 * ```sh
 * $ pulumi import cloudflare:index/listItem:ListItem example '<account_id>/<list_id>/<item_id>'
 * ```
 */
class ListItem extends pulumi.CustomResource {
    /**
     * Get an existing ListItem resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ListItem(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ListItem.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ListItem.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["asn"] = state?.asn;
            resourceInputs["comment"] = state?.comment;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["hostname"] = state?.hostname;
            resourceInputs["ip"] = state?.ip;
            resourceInputs["listId"] = state?.listId;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["operationId"] = state?.operationId;
            resourceInputs["redirect"] = state?.redirect;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.listId === undefined && !opts.urn) {
                throw new Error("Missing required property 'listId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["asn"] = args?.asn;
            resourceInputs["comment"] = args?.comment;
            resourceInputs["hostname"] = args?.hostname;
            resourceInputs["ip"] = args?.ip;
            resourceInputs["listId"] = args?.listId;
            resourceInputs["redirect"] = args?.redirect;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
            resourceInputs["operationId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ListItem.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ListItem = ListItem;
/** @internal */
ListItem.__pulumiType = 'cloudflare:index/listItem:ListItem';
