// @pulumi/cloudflare/magicNetworkMonitoringRule.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicNetworkMonitoringRule = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleMagicNetworkMonitoringRule = new cloudflare.MagicNetworkMonitoringRule("example_magic_network_monitoring_rule", {
 *     accountId: "6f91088a406011ed95aed352566e8d4c",
 *     duration: "1m",
 *     name: "my_rule_1",
 *     automaticAdvertisement: true,
 *     bandwidth: 1000,
 *     packetThreshold: 10000,
 *     prefixes: ["203.0.113.1/32"],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/magicNetworkMonitoringRule:MagicNetworkMonitoringRule example '<account_id>/<rule_id>'
 * ```
 */
class MagicNetworkMonitoringRule extends pulumi.CustomResource {
    /**
     * Get an existing MagicNetworkMonitoringRule resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MagicNetworkMonitoringRule(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of MagicNetworkMonitoringRule.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MagicNetworkMonitoringRule.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["automaticAdvertisement"] = state?.automaticAdvertisement;
            resourceInputs["bandwidth"] = state?.bandwidth;
            resourceInputs["bandwidthThreshold"] = state?.bandwidthThreshold;
            resourceInputs["duration"] = state?.duration;
            resourceInputs["name"] = state?.name;
            resourceInputs["packetThreshold"] = state?.packetThreshold;
            resourceInputs["prefixMatch"] = state?.prefixMatch;
            resourceInputs["prefixes"] = state?.prefixes;
            resourceInputs["type"] = state?.type;
            resourceInputs["zscoreSensitivity"] = state?.zscoreSensitivity;
            resourceInputs["zscoreTarget"] = state?.zscoreTarget;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["automaticAdvertisement"] = args?.automaticAdvertisement;
            resourceInputs["bandwidth"] = args?.bandwidth;
            resourceInputs["duration"] = args?.duration;
            resourceInputs["name"] = args?.name;
            resourceInputs["packetThreshold"] = args?.packetThreshold;
            resourceInputs["prefixes"] = args?.prefixes;
            resourceInputs["bandwidthThreshold"] = undefined /*out*/;
            resourceInputs["prefixMatch"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["zscoreSensitivity"] = undefined /*out*/;
            resourceInputs["zscoreTarget"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MagicNetworkMonitoringRule.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MagicNetworkMonitoringRule = MagicNetworkMonitoringRule;
/** @internal */
MagicNetworkMonitoringRule.__pulumiType = 'cloudflare:index/magicNetworkMonitoringRule:MagicNetworkMonitoringRule';
