// @pulumi/cloudflare/magicTransitSiteLan.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicTransitSiteLan = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleMagicTransitSiteLan = new cloudflare.MagicTransitSiteLan("example_magic_transit_site_lan", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     siteId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     physport: 1,
 *     haLink: true,
 *     name: "name",
 *     nat: {
 *         staticPrefix: "192.0.2.0/24",
 *     },
 *     routedSubnets: [{
 *         nextHop: "192.0.2.1",
 *         prefix: "192.0.2.0/24",
 *         nat: {
 *             staticPrefix: "192.0.2.0/24",
 *         },
 *     }],
 *     staticAddressing: {
 *         address: "192.0.2.0/24",
 *         dhcpRelay: {
 *             serverAddresses: ["192.0.2.1"],
 *         },
 *         dhcpServer: {
 *             dhcpPoolEnd: "192.0.2.1",
 *             dhcpPoolStart: "192.0.2.1",
 *             dnsServer: "192.0.2.1",
 *             dnsServers: ["192.0.2.1"],
 *             reservations: {
 *                 "00:11:22:33:44:55": "192.0.2.100",
 *                 "AA:BB:CC:DD:EE:FF": "192.168.1.101",
 *             },
 *         },
 *         secondaryAddress: "192.0.2.0/24",
 *         virtualAddress: "192.0.2.0/24",
 *     },
 *     vlanTag: 42,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/magicTransitSiteLan:MagicTransitSiteLan example '<account_id>/<site_id>/<lan_id>'
 * ```
 */
class MagicTransitSiteLan extends pulumi.CustomResource {
    /**
     * Get an existing MagicTransitSiteLan resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MagicTransitSiteLan(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of MagicTransitSiteLan.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MagicTransitSiteLan.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["haLink"] = state?.haLink;
            resourceInputs["name"] = state?.name;
            resourceInputs["nat"] = state?.nat;
            resourceInputs["physport"] = state?.physport;
            resourceInputs["routedSubnets"] = state?.routedSubnets;
            resourceInputs["siteId"] = state?.siteId;
            resourceInputs["staticAddressing"] = state?.staticAddressing;
            resourceInputs["vlanTag"] = state?.vlanTag;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.physport === undefined && !opts.urn) {
                throw new Error("Missing required property 'physport'");
            }
            if (args?.siteId === undefined && !opts.urn) {
                throw new Error("Missing required property 'siteId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["haLink"] = args?.haLink;
            resourceInputs["name"] = args?.name;
            resourceInputs["nat"] = args?.nat;
            resourceInputs["physport"] = args?.physport;
            resourceInputs["routedSubnets"] = args?.routedSubnets;
            resourceInputs["siteId"] = args?.siteId;
            resourceInputs["staticAddressing"] = args?.staticAddressing;
            resourceInputs["vlanTag"] = args?.vlanTag;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MagicTransitSiteLan.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MagicTransitSiteLan = MagicTransitSiteLan;
/** @internal */
MagicTransitSiteLan.__pulumiType = 'cloudflare:index/magicTransitSiteLan:MagicTransitSiteLan';
