// @pulumi/cloudflare/magicWanGreTunnel.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicWanGreTunnel = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleMagicWanGreTunnel = new cloudflare.MagicWanGreTunnel("example_magic_wan_gre_tunnel", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     cloudflareGreEndpoint: "203.0.113.1",
 *     customerGreEndpoint: "203.0.113.1",
 *     interfaceAddress: "192.0.2.0/31",
 *     name: "GRE_1",
 *     automaticReturnRouting: true,
 *     bgp: {
 *         customerAsn: 0,
 *         extraPrefixes: ["string"],
 *         md5Key: "md5_key",
 *     },
 *     description: "Tunnel for ISP X",
 *     healthCheck: {
 *         direction: "bidirectional",
 *         enabled: true,
 *         rate: "low",
 *         target: {
 *             saved: "203.0.113.1",
 *         },
 *         type: "request",
 *     },
 *     interfaceAddress6: "2606:54c1:7:0:a9fe:12d2:1:200/127",
 *     mtu: 0,
 *     ttl: 0,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/magicWanGreTunnel:MagicWanGreTunnel example '<account_id>/<gre_tunnel_id>'
 * ```
 */
class MagicWanGreTunnel extends pulumi.CustomResource {
    /**
     * Get an existing MagicWanGreTunnel resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MagicWanGreTunnel(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of MagicWanGreTunnel.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MagicWanGreTunnel.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["cloudflareGreEndpoint"] = state?.cloudflareGreEndpoint;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["customerGreEndpoint"] = state?.customerGreEndpoint;
            resourceInputs["description"] = state?.description;
            resourceInputs["healthCheck"] = state?.healthCheck;
            resourceInputs["interfaceAddress"] = state?.interfaceAddress;
            resourceInputs["interfaceAddress6"] = state?.interfaceAddress6;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["mtu"] = state?.mtu;
            resourceInputs["name"] = state?.name;
            resourceInputs["ttl"] = state?.ttl;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.cloudflareGreEndpoint === undefined && !opts.urn) {
                throw new Error("Missing required property 'cloudflareGreEndpoint'");
            }
            if (args?.customerGreEndpoint === undefined && !opts.urn) {
                throw new Error("Missing required property 'customerGreEndpoint'");
            }
            if (args?.interfaceAddress === undefined && !opts.urn) {
                throw new Error("Missing required property 'interfaceAddress'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["cloudflareGreEndpoint"] = args?.cloudflareGreEndpoint;
            resourceInputs["customerGreEndpoint"] = args?.customerGreEndpoint;
            resourceInputs["description"] = args?.description;
            resourceInputs["healthCheck"] = args?.healthCheck;
            resourceInputs["interfaceAddress"] = args?.interfaceAddress;
            resourceInputs["interfaceAddress6"] = args?.interfaceAddress6;
            resourceInputs["mtu"] = args?.mtu;
            resourceInputs["name"] = args?.name;
            resourceInputs["ttl"] = args?.ttl;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/greTunnel:GreTunnel" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(MagicWanGreTunnel.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MagicWanGreTunnel = MagicWanGreTunnel;
/** @internal */
MagicWanGreTunnel.__pulumiType = 'cloudflare:index/magicWanGreTunnel:MagicWanGreTunnel';
