// @pulumi/cloudflare/magicTransitSite.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.MagicTransitSite = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleMagicTransitSite = new cloudflare.MagicTransitSite("example_magic_transit_site", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     name: "site_1",
 *     connectorId: "ac60d3d0435248289d446cedd870bcf4",
 *     description: "description",
 *     haMode: true,
 *     location: {
 *         lat: "37.6192",
 *         lon: "122.3816",
 *     },
 *     secondaryConnectorId: "8d67040d3835dbcf46ce29da440dc482",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/magicTransitSite:MagicTransitSite example '<account_id>/<site_id>'
 * ```
 */
class MagicTransitSite extends pulumi.CustomResource {
    /**
     * Get an existing MagicTransitSite resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new MagicTransitSite(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of MagicTransitSite.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === MagicTransitSite.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["connectorId"] = state?.connectorId;
            resourceInputs["description"] = state?.description;
            resourceInputs["haMode"] = state?.haMode;
            resourceInputs["location"] = state?.location;
            resourceInputs["name"] = state?.name;
            resourceInputs["secondaryConnectorId"] = state?.secondaryConnectorId;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["connectorId"] = args?.connectorId;
            resourceInputs["description"] = args?.description;
            resourceInputs["haMode"] = args?.haMode;
            resourceInputs["location"] = args?.location;
            resourceInputs["name"] = args?.name;
            resourceInputs["secondaryConnectorId"] = args?.secondaryConnectorId;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(MagicTransitSite.__pulumiType, name, resourceInputs, opts);
    }
}
exports.MagicTransitSite = MagicTransitSite;
/** @internal */
MagicTransitSite.__pulumiType = 'cloudflare:index/magicTransitSite:MagicTransitSite';
