// @pulumi/cloudflare/pagesDomain.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.PagesDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * > A DNS record for the domain is not automatically created. You need to create
 *    a `cloudflareRecord` resource for the domain you want to use.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const examplePagesDomain = new cloudflare.PagesDomain("example_pages_domain", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     projectName: "this-is-my-project-01",
 *     name: "this-is-my-domain-01.com",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/pagesDomain:PagesDomain example '<account_id>/<project_name>/<domain_name>'
 * ```
 */
class PagesDomain extends pulumi.CustomResource {
    /**
     * Get an existing PagesDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new PagesDomain(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of PagesDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === PagesDomain.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["certificateAuthority"] = state?.certificateAuthority;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["domainId"] = state?.domainId;
            resourceInputs["name"] = state?.name;
            resourceInputs["projectName"] = state?.projectName;
            resourceInputs["status"] = state?.status;
            resourceInputs["validationData"] = state?.validationData;
            resourceInputs["verificationData"] = state?.verificationData;
            resourceInputs["zoneTag"] = state?.zoneTag;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.projectName === undefined && !opts.urn) {
                throw new Error("Missing required property 'projectName'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["name"] = args?.name;
            resourceInputs["projectName"] = args?.projectName;
            resourceInputs["certificateAuthority"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["domainId"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["validationData"] = undefined /*out*/;
            resourceInputs["verificationData"] = undefined /*out*/;
            resourceInputs["zoneTag"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(PagesDomain.__pulumiType, name, resourceInputs, opts);
    }
}
exports.PagesDomain = PagesDomain;
/** @internal */
PagesDomain.__pulumiType = 'cloudflare:index/pagesDomain:PagesDomain';
