// @pulumi/cloudflare/queueConsumer.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.QueueConsumer = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleQueueConsumer = new cloudflare.QueueConsumer("example_queue_consumer", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     queueId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     deadLetterQueue: "example-queue",
 *     scriptName: "my-consumer-worker",
 *     settings: {
 *         batchSize: 50,
 *         maxConcurrency: 10,
 *         maxRetries: 3,
 *         maxWaitTimeMs: 5000,
 *         retryDelay: 10,
 *     },
 *     type: "worker",
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class QueueConsumer extends pulumi.CustomResource {
    /**
     * Get an existing QueueConsumer resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new QueueConsumer(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of QueueConsumer.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === QueueConsumer.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["consumerId"] = state?.consumerId;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["deadLetterQueue"] = state?.deadLetterQueue;
            resourceInputs["queueId"] = state?.queueId;
            resourceInputs["script"] = state?.script;
            resourceInputs["scriptName"] = state?.scriptName;
            resourceInputs["settings"] = state?.settings;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.queueId === undefined && !opts.urn) {
                throw new Error("Missing required property 'queueId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["deadLetterQueue"] = args?.deadLetterQueue;
            resourceInputs["queueId"] = args?.queueId;
            resourceInputs["scriptName"] = args?.scriptName;
            resourceInputs["settings"] = args?.settings;
            resourceInputs["type"] = args?.type;
            resourceInputs["consumerId"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["script"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(QueueConsumer.__pulumiType, name, resourceInputs, opts);
    }
}
exports.QueueConsumer = QueueConsumer;
/** @internal */
QueueConsumer.__pulumiType = 'cloudflare:index/queueConsumer:QueueConsumer';
