// @pulumi/cloudflare/r2bucketCors.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.R2BucketCors = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleR2BucketCors = new cloudflare.R2BucketCors("example_r2_bucket_cors", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     bucketName: "example-bucket",
 *     rules: [{
 *         allowed: {
 *             methods: ["GET"],
 *             origins: ["http://localhost:3000"],
 *             headers: ["x-requested-by"],
 *         },
 *         id: "Allow Local Development",
 *         exposeHeaders: ["Content-Encoding"],
 *         maxAgeSeconds: 3600,
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class R2BucketCors extends pulumi.CustomResource {
    /**
     * Get an existing R2BucketCors resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new R2BucketCors(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of R2BucketCors.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === R2BucketCors.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["bucketName"] = state?.bucketName;
            resourceInputs["jurisdiction"] = state?.jurisdiction;
            resourceInputs["rules"] = state?.rules;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.bucketName === undefined && !opts.urn) {
                throw new Error("Missing required property 'bucketName'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["bucketName"] = args?.bucketName;
            resourceInputs["jurisdiction"] = args?.jurisdiction;
            resourceInputs["rules"] = args?.rules;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(R2BucketCors.__pulumiType, name, resourceInputs, opts);
    }
}
exports.R2BucketCors = R2BucketCors;
/** @internal */
R2BucketCors.__pulumiType = 'cloudflare:index/r2BucketCors:R2BucketCors';
