// @pulumi/cloudflare/r2managedDomain.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.R2ManagedDomain = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleR2ManagedDomain = new cloudflare.R2ManagedDomain("example_r2_managed_domain", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     bucketName: "example-bucket",
 *     enabled: true,
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class R2ManagedDomain extends pulumi.CustomResource {
    /**
     * Get an existing R2ManagedDomain resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new R2ManagedDomain(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of R2ManagedDomain.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === R2ManagedDomain.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["bucketId"] = state?.bucketId;
            resourceInputs["bucketName"] = state?.bucketName;
            resourceInputs["domain"] = state?.domain;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["jurisdiction"] = state?.jurisdiction;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.bucketName === undefined && !opts.urn) {
                throw new Error("Missing required property 'bucketName'");
            }
            if (args?.enabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["bucketName"] = args?.bucketName;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["jurisdiction"] = args?.jurisdiction;
            resourceInputs["bucketId"] = undefined /*out*/;
            resourceInputs["domain"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(R2ManagedDomain.__pulumiType, name, resourceInputs, opts);
    }
}
exports.R2ManagedDomain = R2ManagedDomain;
/** @internal */
R2ManagedDomain.__pulumiType = 'cloudflare:index/r2ManagedDomain:R2ManagedDomain';
