// @pulumi/cloudflare/rateLimit.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.RateLimit = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * > `cloudflare.RateLimit` is in a deprecation phase until June 15th, 2025.
 *   During this time period, this resource is still
 *   fully supported but you are strongly advised to move to the
 *   `cloudflare.Ruleset` resource. Full details can be found in the
 *   developer documentation.
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleRateLimit = new cloudflare.RateLimit("example_rate_limit", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     action: {
 *         mode: "challenge",
 *         response: {
 *             body: "<error>This request has been rate-limited.</error>",
 *             contentType: "text/xml",
 *         },
 *         timeout: 86400,
 *     },
 *     match: {
 *         headers: [{
 *             name: "Cf-Cache-Status",
 *             op: "ne",
 *             value: "HIT",
 *         }],
 *         request: {
 *             methods: [
 *                 "GET",
 *                 "POST",
 *             ],
 *             schemes: [
 *                 "HTTP",
 *                 "HTTPS",
 *             ],
 *             url: "*.example.org/path*",
 *         },
 *         response: {
 *             originTraffic: true,
 *         },
 *     },
 *     period: 900,
 *     threshold: 60,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/rateLimit:RateLimit example '<zone_id>/<rate_limit_id>'
 * ```
 */
class RateLimit extends pulumi.CustomResource {
    /**
     * Get an existing RateLimit resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new RateLimit(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of RateLimit.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === RateLimit.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["action"] = state?.action;
            resourceInputs["bypasses"] = state?.bypasses;
            resourceInputs["description"] = state?.description;
            resourceInputs["disabled"] = state?.disabled;
            resourceInputs["match"] = state?.match;
            resourceInputs["period"] = state?.period;
            resourceInputs["threshold"] = state?.threshold;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.action === undefined && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if (args?.match === undefined && !opts.urn) {
                throw new Error("Missing required property 'match'");
            }
            if (args?.period === undefined && !opts.urn) {
                throw new Error("Missing required property 'period'");
            }
            if (args?.threshold === undefined && !opts.urn) {
                throw new Error("Missing required property 'threshold'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["action"] = args?.action;
            resourceInputs["match"] = args?.match;
            resourceInputs["period"] = args?.period;
            resourceInputs["threshold"] = args?.threshold;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["bypasses"] = undefined /*out*/;
            resourceInputs["description"] = undefined /*out*/;
            resourceInputs["disabled"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(RateLimit.__pulumiType, name, resourceInputs, opts);
    }
}
exports.RateLimit = RateLimit;
/** @internal */
RateLimit.__pulumiType = 'cloudflare:index/rateLimit:RateLimit';
