// @pulumi/cloudflare/schemaValidationSchemas.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SchemaValidationSchemas = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleSchemaValidationSchemas = new cloudflare.SchemaValidationSchemas("example_schema_validation_schemas", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     kind: "openapi_v3",
 *     name: "petstore schema",
 *     source: "<schema file contents>",
 *     validationEnabled: true,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/schemaValidationSchemas:SchemaValidationSchemas example '<zone_id>/<schema_id>'
 * ```
 */
class SchemaValidationSchemas extends pulumi.CustomResource {
    /**
     * Get an existing SchemaValidationSchemas resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SchemaValidationSchemas(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of SchemaValidationSchemas.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SchemaValidationSchemas.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["kind"] = state?.kind;
            resourceInputs["name"] = state?.name;
            resourceInputs["schemaId"] = state?.schemaId;
            resourceInputs["source"] = state?.source;
            resourceInputs["validationEnabled"] = state?.validationEnabled;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.kind === undefined && !opts.urn) {
                throw new Error("Missing required property 'kind'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.source === undefined && !opts.urn) {
                throw new Error("Missing required property 'source'");
            }
            if (args?.validationEnabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'validationEnabled'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["kind"] = args?.kind;
            resourceInputs["name"] = args?.name;
            resourceInputs["source"] = args?.source;
            resourceInputs["validationEnabled"] = args?.validationEnabled;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["schemaId"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SchemaValidationSchemas.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SchemaValidationSchemas = SchemaValidationSchemas;
/** @internal */
SchemaValidationSchemas.__pulumiType = 'cloudflare:index/schemaValidationSchemas:SchemaValidationSchemas';
