// @pulumi/cloudflare/snippet.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Snippet = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleSnippet = new cloudflare.Snippet("example_snippet", {
 *     zoneId: "9f1839b6152d298aca64c4e906b6d074",
 *     snippetName: "my_snippet",
 *     files: [{
 *         name: "main.js",
 *         content: `export default {
 *   async fetch(request) {
 *     return new Response('Hello, World!');
 *   }
 * }
 * `,
 *     }],
 *     metadata: {
 *         mainModule: "main.js",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class Snippet extends pulumi.CustomResource {
    /**
     * Get an existing Snippet resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Snippet(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Snippet.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Snippet.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["files"] = state?.files;
            resourceInputs["metadata"] = state?.metadata;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["snippetName"] = state?.snippetName;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.files === undefined && !opts.urn) {
                throw new Error("Missing required property 'files'");
            }
            if (args?.metadata === undefined && !opts.urn) {
                throw new Error("Missing required property 'metadata'");
            }
            if (args?.snippetName === undefined && !opts.urn) {
                throw new Error("Missing required property 'snippetName'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["files"] = args?.files;
            resourceInputs["metadata"] = args?.metadata;
            resourceInputs["snippetName"] = args?.snippetName;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Snippet.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Snippet = Snippet;
/** @internal */
Snippet.__pulumiType = 'cloudflare:index/snippet:Snippet';
