// @pulumi/cloudflare/ssoConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.SsoConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleSsoConnector = new cloudflare.SsoConnector("example_sso_connector", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     emailDomain: "example.com",
 *     beginVerification: true,
 *     useFedrampLanguage: false,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/ssoConnector:SsoConnector example '<account_id>/<sso_connector_id>'
 * ```
 */
class SsoConnector extends pulumi.CustomResource {
    /**
     * Get an existing SsoConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new SsoConnector(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of SsoConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === SsoConnector.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["beginVerification"] = state?.beginVerification;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["emailDomain"] = state?.emailDomain;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["updatedOn"] = state?.updatedOn;
            resourceInputs["useFedrampLanguage"] = state?.useFedrampLanguage;
            resourceInputs["verification"] = state?.verification;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.emailDomain === undefined && !opts.urn) {
                throw new Error("Missing required property 'emailDomain'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["beginVerification"] = args?.beginVerification;
            resourceInputs["emailDomain"] = args?.emailDomain;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["useFedrampLanguage"] = args?.useFedrampLanguage;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["updatedOn"] = undefined /*out*/;
            resourceInputs["verification"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(SsoConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.SsoConnector = SsoConnector;
/** @internal */
SsoConnector.__pulumiType = 'cloudflare:index/ssoConnector:SsoConnector';
