// @pulumi/cloudflare/streamLiveInput.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.StreamLiveInput = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class StreamLiveInput extends pulumi.CustomResource {
    /**
     * Get an existing StreamLiveInput resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new StreamLiveInput(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of StreamLiveInput.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === StreamLiveInput.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["created"] = state?.created;
            resourceInputs["defaultCreator"] = state?.defaultCreator;
            resourceInputs["deleteRecordingAfterDays"] = state?.deleteRecordingAfterDays;
            resourceInputs["liveInputIdentifier"] = state?.liveInputIdentifier;
            resourceInputs["meta"] = state?.meta;
            resourceInputs["modified"] = state?.modified;
            resourceInputs["recording"] = state?.recording;
            resourceInputs["rtmps"] = state?.rtmps;
            resourceInputs["rtmpsPlayback"] = state?.rtmpsPlayback;
            resourceInputs["srt"] = state?.srt;
            resourceInputs["srtPlayback"] = state?.srtPlayback;
            resourceInputs["status"] = state?.status;
            resourceInputs["uid"] = state?.uid;
            resourceInputs["webRtc"] = state?.webRtc;
            resourceInputs["webRtcPlayback"] = state?.webRtcPlayback;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["defaultCreator"] = args?.defaultCreator;
            resourceInputs["deleteRecordingAfterDays"] = args?.deleteRecordingAfterDays;
            resourceInputs["liveInputIdentifier"] = args?.liveInputIdentifier;
            resourceInputs["meta"] = args?.meta;
            resourceInputs["recording"] = args?.recording;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["modified"] = undefined /*out*/;
            resourceInputs["rtmps"] = undefined /*out*/;
            resourceInputs["rtmpsPlayback"] = undefined /*out*/;
            resourceInputs["srt"] = undefined /*out*/;
            resourceInputs["srtPlayback"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["webRtc"] = undefined /*out*/;
            resourceInputs["webRtcPlayback"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(StreamLiveInput.__pulumiType, name, resourceInputs, opts);
    }
}
exports.StreamLiveInput = StreamLiveInput;
/** @internal */
StreamLiveInput.__pulumiType = 'cloudflare:index/streamLiveInput:StreamLiveInput';
