// @pulumi/cloudflare/stream.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Stream = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleStream = new cloudflare.Stream("example_stream", {accountId: "023e105f4ecef8ad9ca31a8372d0c353"});
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class Stream extends pulumi.CustomResource {
    /**
     * Get an existing Stream resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Stream(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Stream.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Stream.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["allowedOrigins"] = state?.allowedOrigins;
            resourceInputs["created"] = state?.created;
            resourceInputs["creator"] = state?.creator;
            resourceInputs["duration"] = state?.duration;
            resourceInputs["identifier"] = state?.identifier;
            resourceInputs["input"] = state?.input;
            resourceInputs["liveInput"] = state?.liveInput;
            resourceInputs["maxDurationSeconds"] = state?.maxDurationSeconds;
            resourceInputs["meta"] = state?.meta;
            resourceInputs["modified"] = state?.modified;
            resourceInputs["playback"] = state?.playback;
            resourceInputs["preview"] = state?.preview;
            resourceInputs["readyToStream"] = state?.readyToStream;
            resourceInputs["readyToStreamAt"] = state?.readyToStreamAt;
            resourceInputs["requireSignedUrls"] = state?.requireSignedUrls;
            resourceInputs["scheduledDeletion"] = state?.scheduledDeletion;
            resourceInputs["size"] = state?.size;
            resourceInputs["status"] = state?.status;
            resourceInputs["thumbnail"] = state?.thumbnail;
            resourceInputs["thumbnailTimestampPct"] = state?.thumbnailTimestampPct;
            resourceInputs["uid"] = state?.uid;
            resourceInputs["uploadExpiry"] = state?.uploadExpiry;
            resourceInputs["uploaded"] = state?.uploaded;
            resourceInputs["watermark"] = state?.watermark;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["allowedOrigins"] = args?.allowedOrigins;
            resourceInputs["creator"] = args?.creator;
            resourceInputs["identifier"] = args?.identifier;
            resourceInputs["maxDurationSeconds"] = args?.maxDurationSeconds;
            resourceInputs["meta"] = args?.meta;
            resourceInputs["requireSignedUrls"] = args?.requireSignedUrls;
            resourceInputs["scheduledDeletion"] = args?.scheduledDeletion;
            resourceInputs["thumbnailTimestampPct"] = args?.thumbnailTimestampPct;
            resourceInputs["uploadExpiry"] = args?.uploadExpiry;
            resourceInputs["created"] = undefined /*out*/;
            resourceInputs["duration"] = undefined /*out*/;
            resourceInputs["input"] = undefined /*out*/;
            resourceInputs["liveInput"] = undefined /*out*/;
            resourceInputs["modified"] = undefined /*out*/;
            resourceInputs["playback"] = undefined /*out*/;
            resourceInputs["preview"] = undefined /*out*/;
            resourceInputs["readyToStream"] = undefined /*out*/;
            resourceInputs["readyToStreamAt"] = undefined /*out*/;
            resourceInputs["size"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["thumbnail"] = undefined /*out*/;
            resourceInputs["uid"] = undefined /*out*/;
            resourceInputs["uploaded"] = undefined /*out*/;
            resourceInputs["watermark"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Stream.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Stream = Stream;
/** @internal */
Stream.__pulumiType = 'cloudflare:index/stream:Stream';
