// @pulumi/cloudflare/tokenValidationRules.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.TokenValidationRules = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleTokenValidationRules = new cloudflare.TokenValidationRules("example_token_validation_rules", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     action: "log",
 *     description: "Long description for Token Validation Rule",
 *     enabled: true,
 *     expression: "is_jwt_valid(\"52973293-cb04-4a97-8f55-e7d2ad1107dd\") or is_jwt_valid(\"46eab8d1-6376-45e3-968f-2c649d77d423\")",
 *     selector: {
 *         excludes: [{
 *             operationIds: [
 *                 "f9c5615e-fe15-48ce-bec6-cfc1946f1bec",
 *                 "56828eae-035a-4396-ba07-51c66d680a04",
 *             ],
 *         }],
 *         includes: [{
 *             host: [
 *                 "v1.example.com",
 *                 "v2.example.com",
 *             ],
 *         }],
 *     },
 *     title: "Example Token Validation Rule",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/tokenValidationRules:TokenValidationRules example '<zone_id>/<rule_id>'
 * ```
 */
class TokenValidationRules extends pulumi.CustomResource {
    /**
     * Get an existing TokenValidationRules resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new TokenValidationRules(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of TokenValidationRules.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === TokenValidationRules.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["action"] = state?.action;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["description"] = state?.description;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["expression"] = state?.expression;
            resourceInputs["lastUpdated"] = state?.lastUpdated;
            resourceInputs["position"] = state?.position;
            resourceInputs["selector"] = state?.selector;
            resourceInputs["title"] = state?.title;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.action === undefined && !opts.urn) {
                throw new Error("Missing required property 'action'");
            }
            if (args?.description === undefined && !opts.urn) {
                throw new Error("Missing required property 'description'");
            }
            if (args?.enabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if (args?.expression === undefined && !opts.urn) {
                throw new Error("Missing required property 'expression'");
            }
            if (args?.selector === undefined && !opts.urn) {
                throw new Error("Missing required property 'selector'");
            }
            if (args?.title === undefined && !opts.urn) {
                throw new Error("Missing required property 'title'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["action"] = args?.action;
            resourceInputs["description"] = args?.description;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["expression"] = args?.expression;
            resourceInputs["position"] = args?.position;
            resourceInputs["selector"] = args?.selector;
            resourceInputs["title"] = args?.title;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["lastUpdated"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(TokenValidationRules.__pulumiType, name, resourceInputs, opts);
    }
}
exports.TokenValidationRules = TokenValidationRules;
/** @internal */
TokenValidationRules.__pulumiType = 'cloudflare:index/tokenValidationRules:TokenValidationRules';
