// @pulumi/cloudflare/waitingRoomEvent.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WaitingRoomEvent = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleWaitingRoomEvent = new cloudflare.WaitingRoomEvent("example_waiting_room_event", {
 *     zoneId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     waitingRoomId: "699d98642c564d2e855e9661899b7252",
 *     eventEndTime: "2021-09-28T17:00:00.000Z",
 *     eventStartTime: "2021-09-28T15:30:00.000Z",
 *     name: "production_webinar_event",
 *     customPageHtml: "{{#waitTimeKnown}} {{waitTime}} mins {{/waitTimeKnown}} {{^waitTimeKnown}} Event is prequeueing / Queue all enabled {{/waitTimeKnown}}",
 *     description: "Production event - DO NOT MODIFY",
 *     disableSessionRenewal: true,
 *     newUsersPerMinute: 200,
 *     prequeueStartTime: "2021-09-28T15:00:00.000Z",
 *     queueingMethod: "random",
 *     sessionDuration: 1,
 *     shuffleAtEventStart: true,
 *     suspended: true,
 *     totalActiveUsers: 200,
 *     turnstileAction: "log",
 *     turnstileMode: "off",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/waitingRoomEvent:WaitingRoomEvent example '<zone_id>/<waiting_room_id>/<event_id>'
 * ```
 */
class WaitingRoomEvent extends pulumi.CustomResource {
    /**
     * Get an existing WaitingRoomEvent resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new WaitingRoomEvent(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of WaitingRoomEvent.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WaitingRoomEvent.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["customPageHtml"] = state?.customPageHtml;
            resourceInputs["description"] = state?.description;
            resourceInputs["disableSessionRenewal"] = state?.disableSessionRenewal;
            resourceInputs["eventEndTime"] = state?.eventEndTime;
            resourceInputs["eventStartTime"] = state?.eventStartTime;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["name"] = state?.name;
            resourceInputs["newUsersPerMinute"] = state?.newUsersPerMinute;
            resourceInputs["prequeueStartTime"] = state?.prequeueStartTime;
            resourceInputs["queueingMethod"] = state?.queueingMethod;
            resourceInputs["sessionDuration"] = state?.sessionDuration;
            resourceInputs["shuffleAtEventStart"] = state?.shuffleAtEventStart;
            resourceInputs["suspended"] = state?.suspended;
            resourceInputs["totalActiveUsers"] = state?.totalActiveUsers;
            resourceInputs["turnstileAction"] = state?.turnstileAction;
            resourceInputs["turnstileMode"] = state?.turnstileMode;
            resourceInputs["waitingRoomId"] = state?.waitingRoomId;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.eventEndTime === undefined && !opts.urn) {
                throw new Error("Missing required property 'eventEndTime'");
            }
            if (args?.eventStartTime === undefined && !opts.urn) {
                throw new Error("Missing required property 'eventStartTime'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.waitingRoomId === undefined && !opts.urn) {
                throw new Error("Missing required property 'waitingRoomId'");
            }
            if (args?.zoneId === undefined && !opts.urn) {
                throw new Error("Missing required property 'zoneId'");
            }
            resourceInputs["customPageHtml"] = args?.customPageHtml;
            resourceInputs["description"] = args?.description;
            resourceInputs["disableSessionRenewal"] = args?.disableSessionRenewal;
            resourceInputs["eventEndTime"] = args?.eventEndTime;
            resourceInputs["eventStartTime"] = args?.eventStartTime;
            resourceInputs["name"] = args?.name;
            resourceInputs["newUsersPerMinute"] = args?.newUsersPerMinute;
            resourceInputs["prequeueStartTime"] = args?.prequeueStartTime;
            resourceInputs["queueingMethod"] = args?.queueingMethod;
            resourceInputs["sessionDuration"] = args?.sessionDuration;
            resourceInputs["shuffleAtEventStart"] = args?.shuffleAtEventStart;
            resourceInputs["suspended"] = args?.suspended;
            resourceInputs["totalActiveUsers"] = args?.totalActiveUsers;
            resourceInputs["turnstileAction"] = args?.turnstileAction;
            resourceInputs["turnstileMode"] = args?.turnstileMode;
            resourceInputs["waitingRoomId"] = args?.waitingRoomId;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(WaitingRoomEvent.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WaitingRoomEvent = WaitingRoomEvent;
/** @internal */
WaitingRoomEvent.__pulumiType = 'cloudflare:index/waitingRoomEvent:WaitingRoomEvent';
