// @pulumi/cloudflare/worker.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.Worker = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleWorker = new cloudflare.Worker("example_worker", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     name: "my-worker",
 *     logpush: true,
 *     observability: {
 *         enabled: true,
 *         headSamplingRate: 1,
 *         logs: {
 *             enabled: true,
 *             headSamplingRate: 1,
 *             invocationLogs: true,
 *         },
 *     },
 *     subdomain: {
 *         enabled: true,
 *         previewsEnabled: true,
 *     },
 *     tags: [
 *         "my-team",
 *         "my-public-api",
 *     ],
 *     tailConsumers: [{
 *         name: "my-tail-consumer",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/worker:Worker example '<account_id>/<worker_id>'
 * ```
 */
class Worker extends pulumi.CustomResource {
    /**
     * Get an existing Worker resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new Worker(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of Worker.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === Worker.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["logpush"] = state?.logpush;
            resourceInputs["name"] = state?.name;
            resourceInputs["observability"] = state?.observability;
            resourceInputs["references"] = state?.references;
            resourceInputs["subdomain"] = state?.subdomain;
            resourceInputs["tags"] = state?.tags;
            resourceInputs["tailConsumers"] = state?.tailConsumers;
            resourceInputs["updatedOn"] = state?.updatedOn;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["logpush"] = args?.logpush;
            resourceInputs["name"] = args?.name;
            resourceInputs["observability"] = args?.observability;
            resourceInputs["subdomain"] = args?.subdomain;
            resourceInputs["tags"] = args?.tags;
            resourceInputs["tailConsumers"] = args?.tailConsumers;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["references"] = undefined /*out*/;
            resourceInputs["updatedOn"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(Worker.__pulumiType, name, resourceInputs, opts);
    }
}
exports.Worker = Worker;
/** @internal */
Worker.__pulumiType = 'cloudflare:index/worker:Worker';
