// @pulumi/cloudflare/workersDeployment.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkersDeployment = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleWorkersDeployment = new cloudflare.WorkersDeployment("example_workers_deployment", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     scriptName: "this-is_my_script-01",
 *     strategy: "percentage",
 *     versions: [{
 *         percentage: 100,
 *         versionId: "182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e",
 *     }],
 *     annotations: {
 *         workersMessage: "Deploy bug fix.",
 *     },
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/workersDeployment:WorkersDeployment example '<account_id>/<script_name>/<deployment_id>'
 * ```
 */
class WorkersDeployment extends pulumi.CustomResource {
    /**
     * Get an existing WorkersDeployment resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new WorkersDeployment(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of WorkersDeployment.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkersDeployment.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["annotations"] = state?.annotations;
            resourceInputs["authorEmail"] = state?.authorEmail;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["scriptName"] = state?.scriptName;
            resourceInputs["source"] = state?.source;
            resourceInputs["strategy"] = state?.strategy;
            resourceInputs["versions"] = state?.versions;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.scriptName === undefined && !opts.urn) {
                throw new Error("Missing required property 'scriptName'");
            }
            if (args?.strategy === undefined && !opts.urn) {
                throw new Error("Missing required property 'strategy'");
            }
            if (args?.versions === undefined && !opts.urn) {
                throw new Error("Missing required property 'versions'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["annotations"] = args?.annotations;
            resourceInputs["scriptName"] = args?.scriptName;
            resourceInputs["strategy"] = args?.strategy;
            resourceInputs["versions"] = args?.versions;
            resourceInputs["authorEmail"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["source"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(WorkersDeployment.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkersDeployment = WorkersDeployment;
/** @internal */
WorkersDeployment.__pulumiType = 'cloudflare:index/workersDeployment:WorkersDeployment';
