// @pulumi/cloudflare/workersForPlatformsNamespace.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkersForPlatformsNamespace = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleWorkersForPlatformsDispatchNamespace = new cloudflare.WorkersForPlatformsDispatchNamespace("example_workers_for_platforms_dispatch_namespace", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     name: "my-dispatch-namespace",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/workersForPlatformsNamespace:WorkersForPlatformsNamespace example '<account_id>/<dispatch_namespace>'
 * ```
 *
 * @deprecated cloudflare.index/workersforplatformsnamespace.WorkersForPlatformsNamespace has been deprecated in favor of cloudflare.index/workersforplatformsdispatchnamespace.WorkersForPlatformsDispatchNamespace
 */
class WorkersForPlatformsNamespace extends pulumi.CustomResource {
    /**
     * Get an existing WorkersForPlatformsNamespace resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        pulumi.log.warn("WorkersForPlatformsNamespace is deprecated: cloudflare.index/workersforplatformsnamespace.WorkersForPlatformsNamespace has been deprecated in favor of cloudflare.index/workersforplatformsdispatchnamespace.WorkersForPlatformsDispatchNamespace");
        return new WorkersForPlatformsNamespace(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of WorkersForPlatformsNamespace.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkersForPlatformsNamespace.__pulumiType;
    }
    /** @deprecated cloudflare.index/workersforplatformsnamespace.WorkersForPlatformsNamespace has been deprecated in favor of cloudflare.index/workersforplatformsdispatchnamespace.WorkersForPlatformsDispatchNamespace */
    constructor(name, argsOrState, opts) {
        pulumi.log.warn("WorkersForPlatformsNamespace is deprecated: cloudflare.index/workersforplatformsnamespace.WorkersForPlatformsNamespace has been deprecated in favor of cloudflare.index/workersforplatformsdispatchnamespace.WorkersForPlatformsDispatchNamespace");
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["createdBy"] = state?.createdBy;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["modifiedBy"] = state?.modifiedBy;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["name"] = state?.name;
            resourceInputs["namespaceId"] = state?.namespaceId;
            resourceInputs["namespaceName"] = state?.namespaceName;
            resourceInputs["scriptCount"] = state?.scriptCount;
            resourceInputs["trustedWorkers"] = state?.trustedWorkers;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["name"] = args?.name;
            resourceInputs["createdBy"] = undefined /*out*/;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["modifiedBy"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
            resourceInputs["namespaceId"] = undefined /*out*/;
            resourceInputs["namespaceName"] = undefined /*out*/;
            resourceInputs["scriptCount"] = undefined /*out*/;
            resourceInputs["trustedWorkers"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/workersForPlatformsNamespace:WorkersForPlatformsNamespace" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkersForPlatformsNamespace.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkersForPlatformsNamespace = WorkersForPlatformsNamespace;
/** @internal */
WorkersForPlatformsNamespace.__pulumiType = 'cloudflare:index/workersForPlatformsNamespace:WorkersForPlatformsNamespace';
