// @pulumi/cloudflare/workerScript.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.WorkerScript = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * > For more direct control over Workers resources, we recommend the beta `cloudflare.Worker`, `cloudflare.WorkerVersion`, and `cloudflare.WorkersDeployment` resources. See how to use them in the [developer documentation](https://developers.cloudflare.com/workers/platform/infrastructure-as-code/).
 *
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 * import * as std from "@pulumi/std";
 *
 * const exampleWorkersScript = new cloudflare.WorkersScript("example_workers_script", {
 *     accountId: "023e105f4ecef8ad9ca31a8372d0c353",
 *     scriptName: "this-is_my_script-01",
 *     assets: {
 *         config: {
 *             headers: `        /dashboard/*
 *         X-Frame-Options: DENY
 *
 *         /static/*
 *         Access-Control-Allow-Origin: *
 * `,
 *             redirects: `        /foo /bar 301
 *         /news/* /blog/:splat
 * `,
 *             htmlHandling: "auto-trailing-slash",
 *             notFoundHandling: "none",
 *             runWorkerFirst: false,
 *         },
 *         jwt: "jwt",
 *     },
 *     bindings: [{
 *         name: "MY_ENV_VAR",
 *         text: "my_data",
 *         type: "plain_text",
 *     }],
 *     compatibilityDate: "2021-01-01",
 *     compatibilityFlags: ["nodejs_compat"],
 *     contentFile: "worker.js",
 *     contentSha256: std.filesha256({
 *         input: "worker.js",
 *     }).then(invoke => invoke.result),
 *     keepAssets: false,
 *     keepBindings: ["kv_namespace"],
 *     limits: {
 *         cpuMs: 50,
 *     },
 *     logpush: false,
 *     mainModule: "worker.js",
 *     migrations: {
 *         deletedClasses: ["string"],
 *         newClasses: ["string"],
 *         newSqliteClasses: ["string"],
 *         newTag: "v2",
 *         oldTag: "v1",
 *         renamedClasses: [{
 *             from: "from",
 *             to: "to",
 *         }],
 *         transferredClasses: [{
 *             from: "from",
 *             fromScript: "from_script",
 *             to: "to",
 *         }],
 *     },
 *     observability: {
 *         enabled: true,
 *         headSamplingRate: 0.1,
 *         logs: {
 *             enabled: true,
 *             invocationLogs: true,
 *             destinations: ["cloudflare"],
 *             headSamplingRate: 0.1,
 *             persist: true,
 *         },
 *     },
 *     placement: {
 *         mode: "smart",
 *     },
 *     tailConsumers: [{
 *         service: "my-log-consumer",
 *         environment: "production",
 *         namespace: "my-namespace",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/workerScript:WorkerScript example '<account_id>/<script_name>'
 * ```
 *
 * @deprecated cloudflare.index/workerscript.WorkerScript has been deprecated in favor of cloudflare.index/workersscript.WorkersScript
 */
class WorkerScript extends pulumi.CustomResource {
    /**
     * Get an existing WorkerScript resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        pulumi.log.warn("WorkerScript is deprecated: cloudflare.index/workerscript.WorkerScript has been deprecated in favor of cloudflare.index/workersscript.WorkersScript");
        return new WorkerScript(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of WorkerScript.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === WorkerScript.__pulumiType;
    }
    /** @deprecated cloudflare.index/workerscript.WorkerScript has been deprecated in favor of cloudflare.index/workersscript.WorkersScript */
    constructor(name, argsOrState, opts) {
        pulumi.log.warn("WorkerScript is deprecated: cloudflare.index/workerscript.WorkerScript has been deprecated in favor of cloudflare.index/workersscript.WorkersScript");
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["assets"] = state?.assets;
            resourceInputs["bindings"] = state?.bindings;
            resourceInputs["bodyPart"] = state?.bodyPart;
            resourceInputs["compatibilityDate"] = state?.compatibilityDate;
            resourceInputs["compatibilityFlags"] = state?.compatibilityFlags;
            resourceInputs["content"] = state?.content;
            resourceInputs["contentFile"] = state?.contentFile;
            resourceInputs["contentSha256"] = state?.contentSha256;
            resourceInputs["contentType"] = state?.contentType;
            resourceInputs["createdOn"] = state?.createdOn;
            resourceInputs["etag"] = state?.etag;
            resourceInputs["handlers"] = state?.handlers;
            resourceInputs["hasAssets"] = state?.hasAssets;
            resourceInputs["hasModules"] = state?.hasModules;
            resourceInputs["keepAssets"] = state?.keepAssets;
            resourceInputs["keepBindings"] = state?.keepBindings;
            resourceInputs["lastDeployedFrom"] = state?.lastDeployedFrom;
            resourceInputs["limits"] = state?.limits;
            resourceInputs["logpush"] = state?.logpush;
            resourceInputs["mainModule"] = state?.mainModule;
            resourceInputs["migrationTag"] = state?.migrationTag;
            resourceInputs["migrations"] = state?.migrations;
            resourceInputs["modifiedOn"] = state?.modifiedOn;
            resourceInputs["namedHandlers"] = state?.namedHandlers;
            resourceInputs["observability"] = state?.observability;
            resourceInputs["placement"] = state?.placement;
            resourceInputs["scriptName"] = state?.scriptName;
            resourceInputs["startupTimeMs"] = state?.startupTimeMs;
            resourceInputs["tailConsumers"] = state?.tailConsumers;
            resourceInputs["usageModel"] = state?.usageModel;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.scriptName === undefined && !opts.urn) {
                throw new Error("Missing required property 'scriptName'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["assets"] = args?.assets;
            resourceInputs["bindings"] = args?.bindings;
            resourceInputs["bodyPart"] = args?.bodyPart;
            resourceInputs["compatibilityDate"] = args?.compatibilityDate;
            resourceInputs["compatibilityFlags"] = args?.compatibilityFlags;
            resourceInputs["content"] = args?.content;
            resourceInputs["contentFile"] = args?.contentFile;
            resourceInputs["contentSha256"] = args?.contentSha256;
            resourceInputs["contentType"] = args?.contentType;
            resourceInputs["keepAssets"] = args?.keepAssets;
            resourceInputs["keepBindings"] = args?.keepBindings;
            resourceInputs["limits"] = args?.limits;
            resourceInputs["logpush"] = args?.logpush;
            resourceInputs["mainModule"] = args?.mainModule;
            resourceInputs["migrations"] = args?.migrations;
            resourceInputs["observability"] = args?.observability;
            resourceInputs["placement"] = args?.placement;
            resourceInputs["scriptName"] = args?.scriptName;
            resourceInputs["tailConsumers"] = args?.tailConsumers;
            resourceInputs["usageModel"] = args?.usageModel;
            resourceInputs["createdOn"] = undefined /*out*/;
            resourceInputs["etag"] = undefined /*out*/;
            resourceInputs["handlers"] = undefined /*out*/;
            resourceInputs["hasAssets"] = undefined /*out*/;
            resourceInputs["hasModules"] = undefined /*out*/;
            resourceInputs["lastDeployedFrom"] = undefined /*out*/;
            resourceInputs["migrationTag"] = undefined /*out*/;
            resourceInputs["modifiedOn"] = undefined /*out*/;
            resourceInputs["namedHandlers"] = undefined /*out*/;
            resourceInputs["startupTimeMs"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/workerScript:WorkerScript" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(WorkerScript.__pulumiType, name, resourceInputs, opts);
    }
}
exports.WorkerScript = WorkerScript;
/** @internal */
WorkerScript.__pulumiType = 'cloudflare:index/workerScript:WorkerScript';
