// @pulumi/cloudflare/zeroTrustAccessMtlsHostnameSettings.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustAccessMtlsHostnameSettings = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustAccessMtlsHostnameSettings = new cloudflare.ZeroTrustAccessMtlsHostnameSettings("example_zero_trust_access_mtls_hostname_settings", {
 *     settings: [{
 *         chinaNetwork: false,
 *         clientCertificateForwarding: true,
 *         hostname: "admin.example.com",
 *     }],
 *     zoneId: "zone_id",
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class ZeroTrustAccessMtlsHostnameSettings extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustAccessMtlsHostnameSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustAccessMtlsHostnameSettings(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustAccessMtlsHostnameSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustAccessMtlsHostnameSettings.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["chinaNetwork"] = state?.chinaNetwork;
            resourceInputs["clientCertificateForwarding"] = state?.clientCertificateForwarding;
            resourceInputs["hostname"] = state?.hostname;
            resourceInputs["settings"] = state?.settings;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.settings === undefined && !opts.urn) {
                throw new Error("Missing required property 'settings'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["settings"] = args?.settings;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["chinaNetwork"] = undefined /*out*/;
            resourceInputs["clientCertificateForwarding"] = undefined /*out*/;
            resourceInputs["hostname"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/accessMutualTlsHostnameSettings:AccessMutualTlsHostnameSettings" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ZeroTrustAccessMtlsHostnameSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustAccessMtlsHostnameSettings = ZeroTrustAccessMtlsHostnameSettings;
/** @internal */
ZeroTrustAccessMtlsHostnameSettings.__pulumiType = 'cloudflare:index/zeroTrustAccessMtlsHostnameSettings:ZeroTrustAccessMtlsHostnameSettings';
