// @pulumi/cloudflare/zeroTrustTunnelCloudflaredVirtualNetwork.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustTunnelCloudflaredVirtualNetwork = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustTunnelCloudflaredVirtualNetwork = new cloudflare.ZeroTrustTunnelCloudflaredVirtualNetwork("example_zero_trust_tunnel_cloudflared_virtual_network", {
 *     accountId: "699d98642c564d2e855e9661899b7252",
 *     name: "us-east-1-vpc",
 *     comment: "Staging VPC for data science",
 *     isDefault: true,
 *     isDefaultNetwork: false,
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustTunnelCloudflaredVirtualNetwork:ZeroTrustTunnelCloudflaredVirtualNetwork example '<account_id>/<virtual_network_id>'
 * ```
 */
class ZeroTrustTunnelCloudflaredVirtualNetwork extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustTunnelCloudflaredVirtualNetwork resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustTunnelCloudflaredVirtualNetwork(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustTunnelCloudflaredVirtualNetwork.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustTunnelCloudflaredVirtualNetwork.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["comment"] = state?.comment;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["deletedAt"] = state?.deletedAt;
            resourceInputs["isDefault"] = state?.isDefault;
            resourceInputs["isDefaultNetwork"] = state?.isDefaultNetwork;
            resourceInputs["name"] = state?.name;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["comment"] = args?.comment;
            resourceInputs["isDefault"] = args?.isDefault;
            resourceInputs["isDefaultNetwork"] = args?.isDefaultNetwork;
            resourceInputs["name"] = args?.name;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["deletedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/tunnelVirtualNetwork:TunnelVirtualNetwork" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ZeroTrustTunnelCloudflaredVirtualNetwork.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustTunnelCloudflaredVirtualNetwork = ZeroTrustTunnelCloudflaredVirtualNetwork;
/** @internal */
ZeroTrustTunnelCloudflaredVirtualNetwork.__pulumiType = 'cloudflare:index/zeroTrustTunnelCloudflaredVirtualNetwork:ZeroTrustTunnelCloudflaredVirtualNetwork';
