// @pulumi/cloudflare/zeroTrustDlpDataset.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustDlpDataset = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustDlpDataset = new cloudflare.ZeroTrustDlpDataset("example_zero_trust_dlp_dataset", {
 *     accountId: "account_id",
 *     name: "name",
 *     caseSensitive: true,
 *     description: "description",
 *     encodingVersion: 0,
 *     secret: true,
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class ZeroTrustDlpDataset extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustDlpDataset resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustDlpDataset(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustDlpDataset.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustDlpDataset.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["caseSensitive"] = state?.caseSensitive;
            resourceInputs["columns"] = state?.columns;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["dataset"] = state?.dataset;
            resourceInputs["datasetId"] = state?.datasetId;
            resourceInputs["description"] = state?.description;
            resourceInputs["encodingVersion"] = state?.encodingVersion;
            resourceInputs["maxCells"] = state?.maxCells;
            resourceInputs["name"] = state?.name;
            resourceInputs["numCells"] = state?.numCells;
            resourceInputs["secret"] = state?.secret;
            resourceInputs["status"] = state?.status;
            resourceInputs["updatedAt"] = state?.updatedAt;
            resourceInputs["uploads"] = state?.uploads;
            resourceInputs["version"] = state?.version;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["caseSensitive"] = args?.caseSensitive;
            resourceInputs["datasetId"] = args?.datasetId;
            resourceInputs["description"] = args?.description;
            resourceInputs["encodingVersion"] = args?.encodingVersion;
            resourceInputs["name"] = args?.name;
            resourceInputs["secret"] = args?.secret;
            resourceInputs["columns"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["dataset"] = undefined /*out*/;
            resourceInputs["maxCells"] = undefined /*out*/;
            resourceInputs["numCells"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
            resourceInputs["uploads"] = undefined /*out*/;
            resourceInputs["version"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ZeroTrustDlpDataset.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustDlpDataset = ZeroTrustDlpDataset;
/** @internal */
ZeroTrustDlpDataset.__pulumiType = 'cloudflare:index/zeroTrustDlpDataset:ZeroTrustDlpDataset';
