// @pulumi/cloudflare/zeroTrustDlpEntry.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustDlpEntry = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustDlpEntry = new cloudflare.ZeroTrustDlpEntry("example_zero_trust_dlp_entry", {
 *     accountId: "account_id",
 *     enabled: true,
 *     name: "name",
 *     pattern: {
 *         regex: "regex",
 *         validation: "luhn",
 *     },
 *     profileId: "182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustDlpEntry:ZeroTrustDlpEntry example '<account_id>/<entry_id>'
 * ```
 */
class ZeroTrustDlpEntry extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustDlpEntry resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustDlpEntry(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustDlpEntry.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustDlpEntry.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["caseSensitive"] = state?.caseSensitive;
            resourceInputs["confidence"] = state?.confidence;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["enabled"] = state?.enabled;
            resourceInputs["name"] = state?.name;
            resourceInputs["pattern"] = state?.pattern;
            resourceInputs["profileId"] = state?.profileId;
            resourceInputs["profiles"] = state?.profiles;
            resourceInputs["secret"] = state?.secret;
            resourceInputs["type"] = state?.type;
            resourceInputs["updatedAt"] = state?.updatedAt;
            resourceInputs["uploadStatus"] = state?.uploadStatus;
            resourceInputs["variant"] = state?.variant;
            resourceInputs["wordList"] = state?.wordList;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.enabled === undefined && !opts.urn) {
                throw new Error("Missing required property 'enabled'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.pattern === undefined && !opts.urn) {
                throw new Error("Missing required property 'pattern'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["enabled"] = args?.enabled;
            resourceInputs["name"] = args?.name;
            resourceInputs["pattern"] = args?.pattern;
            resourceInputs["profileId"] = args?.profileId;
            resourceInputs["type"] = args?.type;
            resourceInputs["caseSensitive"] = undefined /*out*/;
            resourceInputs["confidence"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["profiles"] = undefined /*out*/;
            resourceInputs["secret"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
            resourceInputs["uploadStatus"] = undefined /*out*/;
            resourceInputs["variant"] = undefined /*out*/;
            resourceInputs["wordList"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ZeroTrustDlpEntry.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustDlpEntry = ZeroTrustDlpEntry;
/** @internal */
ZeroTrustDlpEntry.__pulumiType = 'cloudflare:index/zeroTrustDlpEntry:ZeroTrustDlpEntry';
