// @pulumi/cloudflare/zeroTrustRiskScoringIntegration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustRiskScoringIntegration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustRiskScoringIntegration = new cloudflare.ZeroTrustRiskScoringIntegration("example_zero_trust_risk_scoring_integration", {
 *     accountId: "account_id",
 *     integrationType: "Okta",
 *     tenantUrl: "https://example.com",
 *     referenceId: "reference_id",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustRiskScoringIntegration:ZeroTrustRiskScoringIntegration example '<account_id>/<integration_id>'
 * ```
 */
class ZeroTrustRiskScoringIntegration extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustRiskScoringIntegration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustRiskScoringIntegration(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustRiskScoringIntegration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustRiskScoringIntegration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["accountTag"] = state?.accountTag;
            resourceInputs["active"] = state?.active;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["integrationType"] = state?.integrationType;
            resourceInputs["referenceId"] = state?.referenceId;
            resourceInputs["tenantUrl"] = state?.tenantUrl;
            resourceInputs["wellKnownUrl"] = state?.wellKnownUrl;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.integrationType === undefined && !opts.urn) {
                throw new Error("Missing required property 'integrationType'");
            }
            if (args?.tenantUrl === undefined && !opts.urn) {
                throw new Error("Missing required property 'tenantUrl'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["active"] = args?.active;
            resourceInputs["integrationType"] = args?.integrationType;
            resourceInputs["referenceId"] = args?.referenceId;
            resourceInputs["tenantUrl"] = args?.tenantUrl;
            resourceInputs["accountTag"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["wellKnownUrl"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ZeroTrustRiskScoringIntegration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustRiskScoringIntegration = ZeroTrustRiskScoringIntegration;
/** @internal */
ZeroTrustRiskScoringIntegration.__pulumiType = 'cloudflare:index/zeroTrustRiskScoringIntegration:ZeroTrustRiskScoringIntegration';
