// @pulumi/cloudflare/zeroTrustAccessServiceToken.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustAccessServiceToken = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustAccessServiceToken = new cloudflare.ZeroTrustAccessServiceToken("example_zero_trust_access_service_token", {
 *     name: "CI/CD token",
 *     zoneId: "zone_id",
 *     clientSecretVersion: 0,
 *     duration: "60m",
 *     previousClientSecretExpiresAt: "2014-01-01T05:20:00.12345Z",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustAccessServiceToken:ZeroTrustAccessServiceToken example '<{accounts|zones}/{account_id|zone_id}>/<service_token_id>'
 * ```
 */
class ZeroTrustAccessServiceToken extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustAccessServiceToken resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustAccessServiceToken(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustAccessServiceToken.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustAccessServiceToken.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["clientId"] = state?.clientId;
            resourceInputs["clientSecret"] = state?.clientSecret;
            resourceInputs["clientSecretVersion"] = state?.clientSecretVersion;
            resourceInputs["duration"] = state?.duration;
            resourceInputs["expiresAt"] = state?.expiresAt;
            resourceInputs["name"] = state?.name;
            resourceInputs["previousClientSecretExpiresAt"] = state?.previousClientSecretExpiresAt;
            resourceInputs["zoneId"] = state?.zoneId;
        }
        else {
            const args = argsOrState;
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["clientSecretVersion"] = args?.clientSecretVersion;
            resourceInputs["duration"] = args?.duration;
            resourceInputs["name"] = args?.name;
            resourceInputs["previousClientSecretExpiresAt"] = args?.previousClientSecretExpiresAt;
            resourceInputs["zoneId"] = args?.zoneId;
            resourceInputs["clientId"] = undefined /*out*/;
            resourceInputs["clientSecret"] = undefined /*out*/;
            resourceInputs["expiresAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/accessServiceToken:AccessServiceToken" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        const secretOpts = { additionalSecretOutputs: ["clientSecret"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(ZeroTrustAccessServiceToken.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustAccessServiceToken = ZeroTrustAccessServiceToken;
/** @internal */
ZeroTrustAccessServiceToken.__pulumiType = 'cloudflare:index/zeroTrustAccessServiceToken:ZeroTrustAccessServiceToken';
