// @pulumi/cloudflare/zeroTrustDevicePostureIntegration.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustDevicePostureIntegration = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustDevicePostureIntegration = new cloudflare.ZeroTrustDevicePostureIntegration("example_zero_trust_device_posture_integration", {
 *     accountId: "699d98642c564d2e855e9661899b7252",
 *     config: {
 *         apiUrl: "https://as123.awmdm.com/API",
 *         authUrl: "https://na.uemauth.vmwservices.com/connect/token",
 *         clientId: "example client id",
 *         clientSecret: "example client secret",
 *     },
 *     interval: "10m",
 *     name: "My Workspace One Integration",
 *     type: "workspace_one",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustDevicePostureIntegration:ZeroTrustDevicePostureIntegration example '<account_id>/<integration_id>'
 * ```
 */
class ZeroTrustDevicePostureIntegration extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustDevicePostureIntegration resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustDevicePostureIntegration(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustDevicePostureIntegration.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustDevicePostureIntegration.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["config"] = state?.config;
            resourceInputs["interval"] = state?.interval;
            resourceInputs["name"] = state?.name;
            resourceInputs["type"] = state?.type;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.config === undefined && !opts.urn) {
                throw new Error("Missing required property 'config'");
            }
            if (args?.interval === undefined && !opts.urn) {
                throw new Error("Missing required property 'interval'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            if (args?.type === undefined && !opts.urn) {
                throw new Error("Missing required property 'type'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["config"] = args?.config;
            resourceInputs["interval"] = args?.interval;
            resourceInputs["name"] = args?.name;
            resourceInputs["type"] = args?.type;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/devicePostureIntegration:DevicePostureIntegration" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ZeroTrustDevicePostureIntegration.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustDevicePostureIntegration = ZeroTrustDevicePostureIntegration;
/** @internal */
ZeroTrustDevicePostureIntegration.__pulumiType = 'cloudflare:index/zeroTrustDevicePostureIntegration:ZeroTrustDevicePostureIntegration';
