// @pulumi/cloudflare/zeroTrustDeviceSettings.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustDeviceSettings = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustDeviceSettings = new cloudflare.ZeroTrustDeviceSettings("example_zero_trust_device_settings", {
 *     accountId: "699d98642c564d2e855e9661899b7252",
 *     disableForTime: 0,
 *     externalEmergencySignalEnabled: true,
 *     externalEmergencySignalFingerprint: "abcd1234567890abcd1234567890abcd1234567890abcd1234567890abcd1234",
 *     externalEmergencySignalInterval: "5m",
 *     externalEmergencySignalUrl: "https://192.0.2.1/signal",
 *     gatewayProxyEnabled: true,
 *     gatewayUdpProxyEnabled: true,
 *     rootCertificateInstallationEnabled: true,
 *     useZtVirtualIp: true,
 * });
 * ```
 *
 * ## Import
 *
 * ~> This resource does not currently support `pulumi import`.
 */
class ZeroTrustDeviceSettings extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustDeviceSettings resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustDeviceSettings(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustDeviceSettings.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustDeviceSettings.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["disableForTime"] = state?.disableForTime;
            resourceInputs["externalEmergencySignalEnabled"] = state?.externalEmergencySignalEnabled;
            resourceInputs["externalEmergencySignalFingerprint"] = state?.externalEmergencySignalFingerprint;
            resourceInputs["externalEmergencySignalInterval"] = state?.externalEmergencySignalInterval;
            resourceInputs["externalEmergencySignalUrl"] = state?.externalEmergencySignalUrl;
            resourceInputs["gatewayProxyEnabled"] = state?.gatewayProxyEnabled;
            resourceInputs["gatewayUdpProxyEnabled"] = state?.gatewayUdpProxyEnabled;
            resourceInputs["rootCertificateInstallationEnabled"] = state?.rootCertificateInstallationEnabled;
            resourceInputs["useZtVirtualIp"] = state?.useZtVirtualIp;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["disableForTime"] = args?.disableForTime;
            resourceInputs["externalEmergencySignalEnabled"] = args?.externalEmergencySignalEnabled;
            resourceInputs["externalEmergencySignalFingerprint"] = args?.externalEmergencySignalFingerprint;
            resourceInputs["externalEmergencySignalInterval"] = args?.externalEmergencySignalInterval;
            resourceInputs["externalEmergencySignalUrl"] = args?.externalEmergencySignalUrl;
            resourceInputs["gatewayProxyEnabled"] = args?.gatewayProxyEnabled;
            resourceInputs["gatewayUdpProxyEnabled"] = args?.gatewayUdpProxyEnabled;
            resourceInputs["rootCertificateInstallationEnabled"] = args?.rootCertificateInstallationEnabled;
            resourceInputs["useZtVirtualIp"] = args?.useZtVirtualIp;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        super(ZeroTrustDeviceSettings.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustDeviceSettings = ZeroTrustDeviceSettings;
/** @internal */
ZeroTrustDeviceSettings.__pulumiType = 'cloudflare:index/zeroTrustDeviceSettings:ZeroTrustDeviceSettings';
