// @pulumi/cloudflare/zeroTrustDlpCustomProfile.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustDlpCustomProfile = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustDlpCustomProfile = new cloudflare.ZeroTrustDlpCustomProfile("example_zero_trust_dlp_custom_profile", {
 *     accountId: "account_id",
 *     name: "name",
 *     aiContextEnabled: true,
 *     allowedMatchCount: 5,
 *     confidenceThreshold: "confidence_threshold",
 *     contextAwareness: {
 *         enabled: true,
 *         skip: {
 *             files: true,
 *         },
 *     },
 *     description: "description",
 *     ocrEnabled: true,
 *     sharedEntries: [{
 *         enabled: true,
 *         entryId: "182bd5e5-6e1a-4fe4-a799-aa6d9a6ab26e",
 *         entryType: "custom",
 *     }],
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustDlpCustomProfile:ZeroTrustDlpCustomProfile example '<account_id>/<profile_id>'
 * ```
 */
class ZeroTrustDlpCustomProfile extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustDlpCustomProfile resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustDlpCustomProfile(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustDlpCustomProfile.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustDlpCustomProfile.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["aiContextEnabled"] = state?.aiContextEnabled;
            resourceInputs["allowedMatchCount"] = state?.allowedMatchCount;
            resourceInputs["confidenceThreshold"] = state?.confidenceThreshold;
            resourceInputs["contextAwareness"] = state?.contextAwareness;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["description"] = state?.description;
            resourceInputs["entries"] = state?.entries;
            resourceInputs["name"] = state?.name;
            resourceInputs["ocrEnabled"] = state?.ocrEnabled;
            resourceInputs["openAccess"] = state?.openAccess;
            resourceInputs["sharedEntries"] = state?.sharedEntries;
            resourceInputs["type"] = state?.type;
            resourceInputs["updatedAt"] = state?.updatedAt;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["aiContextEnabled"] = args?.aiContextEnabled;
            resourceInputs["allowedMatchCount"] = args?.allowedMatchCount;
            resourceInputs["confidenceThreshold"] = args?.confidenceThreshold;
            resourceInputs["contextAwareness"] = args?.contextAwareness;
            resourceInputs["description"] = args?.description;
            resourceInputs["entries"] = args?.entries;
            resourceInputs["name"] = args?.name;
            resourceInputs["ocrEnabled"] = args?.ocrEnabled;
            resourceInputs["sharedEntries"] = args?.sharedEntries;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["openAccess"] = undefined /*out*/;
            resourceInputs["type"] = undefined /*out*/;
            resourceInputs["updatedAt"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const aliasOpts = { aliases: [{ type: "cloudflare:index/dlpCustomProfile:DlpCustomProfile" }] };
        opts = pulumi.mergeOptions(opts, aliasOpts);
        super(ZeroTrustDlpCustomProfile.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustDlpCustomProfile = ZeroTrustDlpCustomProfile;
/** @internal */
ZeroTrustDlpCustomProfile.__pulumiType = 'cloudflare:index/zeroTrustDlpCustomProfile:ZeroTrustDlpCustomProfile';
