// @pulumi/cloudflare/zeroTrustTunnelWarpConnector.js
"use strict";
// *** WARNING: this file was generated by pulumi-language-nodejs. ***
// *** Do not edit by hand unless you're certain you know what you are doing! ***
Object.defineProperty(exports, "__esModule", { value: true });
exports.ZeroTrustTunnelWarpConnector = void 0;
const pulumi = require("@pulumi/pulumi");
const utilities = require("./utilities");
/**
 * ## Example Usage
 *
 * ```typescript
 * import * as pulumi from "@pulumi/pulumi";
 * import * as cloudflare from "@pulumi/cloudflare";
 *
 * const exampleZeroTrustTunnelWarpConnector = new cloudflare.ZeroTrustTunnelWarpConnector("example_zero_trust_tunnel_warp_connector", {
 *     accountId: "699d98642c564d2e855e9661899b7252",
 *     name: "blog",
 * });
 * ```
 *
 * ## Import
 *
 * ```sh
 * $ pulumi import cloudflare:index/zeroTrustTunnelWarpConnector:ZeroTrustTunnelWarpConnector example '<account_id>/<tunnel_id>'
 * ```
 */
class ZeroTrustTunnelWarpConnector extends pulumi.CustomResource {
    /**
     * Get an existing ZeroTrustTunnelWarpConnector resource's state with the given name, ID, and optional extra
     * properties used to qualify the lookup.
     *
     * @param name The _unique_ name of the resulting resource.
     * @param id The _unique_ provider ID of the resource to lookup.
     * @param state Any extra arguments used during the lookup.
     * @param opts Optional settings to control the behavior of the CustomResource.
     */
    static get(name, id, state, opts) {
        return new ZeroTrustTunnelWarpConnector(name, state, { ...opts, id: id });
    }
    /**
     * Returns true if the given object is an instance of ZeroTrustTunnelWarpConnector.  This is designed to work even
     * when multiple copies of the Pulumi SDK have been loaded into the same process.
     */
    static isInstance(obj) {
        if (obj === undefined || obj === null) {
            return false;
        }
        return obj['__pulumiType'] === ZeroTrustTunnelWarpConnector.__pulumiType;
    }
    constructor(name, argsOrState, opts) {
        let resourceInputs = {};
        opts = opts || {};
        if (opts.id) {
            const state = argsOrState;
            resourceInputs["accountId"] = state?.accountId;
            resourceInputs["accountTag"] = state?.accountTag;
            resourceInputs["connections"] = state?.connections;
            resourceInputs["connsActiveAt"] = state?.connsActiveAt;
            resourceInputs["connsInactiveAt"] = state?.connsInactiveAt;
            resourceInputs["createdAt"] = state?.createdAt;
            resourceInputs["deletedAt"] = state?.deletedAt;
            resourceInputs["metadata"] = state?.metadata;
            resourceInputs["name"] = state?.name;
            resourceInputs["status"] = state?.status;
            resourceInputs["tunType"] = state?.tunType;
            resourceInputs["tunnelSecret"] = state?.tunnelSecret;
        }
        else {
            const args = argsOrState;
            if (args?.accountId === undefined && !opts.urn) {
                throw new Error("Missing required property 'accountId'");
            }
            if (args?.name === undefined && !opts.urn) {
                throw new Error("Missing required property 'name'");
            }
            resourceInputs["accountId"] = args?.accountId;
            resourceInputs["name"] = args?.name;
            resourceInputs["tunnelSecret"] = args?.tunnelSecret ? pulumi.secret(args.tunnelSecret) : undefined;
            resourceInputs["accountTag"] = undefined /*out*/;
            resourceInputs["connections"] = undefined /*out*/;
            resourceInputs["connsActiveAt"] = undefined /*out*/;
            resourceInputs["connsInactiveAt"] = undefined /*out*/;
            resourceInputs["createdAt"] = undefined /*out*/;
            resourceInputs["deletedAt"] = undefined /*out*/;
            resourceInputs["metadata"] = undefined /*out*/;
            resourceInputs["status"] = undefined /*out*/;
            resourceInputs["tunType"] = undefined /*out*/;
        }
        opts = pulumi.mergeOptions(utilities.resourceOptsDefaults(), opts);
        const secretOpts = { additionalSecretOutputs: ["tunnelSecret"] };
        opts = pulumi.mergeOptions(opts, secretOpts);
        super(ZeroTrustTunnelWarpConnector.__pulumiType, name, resourceInputs, opts);
    }
}
exports.ZeroTrustTunnelWarpConnector = ZeroTrustTunnelWarpConnector;
/** @internal */
ZeroTrustTunnelWarpConnector.__pulumiType = 'cloudflare:index/zeroTrustTunnelWarpConnector:ZeroTrustTunnelWarpConnector';
